{
  "total_count": 6551,
  "incomplete_results": false,
  "items": [
    {
      "url": "https://api.github.com/repos/JohnLangford/vowpal_wabbit/issues/1472",
      "repository_url": "https://api.github.com/repos/JohnLangford/vowpal_wabbit",
      "labels_url": "https://api.github.com/repos/JohnLangford/vowpal_wabbit/issues/1472/labels{/name}",
      "comments_url": "https://api.github.com/repos/JohnLangford/vowpal_wabbit/issues/1472/comments",
      "events_url": "https://api.github.com/repos/JohnLangford/vowpal_wabbit/issues/1472/events",
      "html_url": "https://github.com/JohnLangford/vowpal_wabbit/issues/1472",
      "id": 318997532,
      "node_id": "MDU6SXNzdWUzMTg5OTc1MzI=",
      "number": 1472,
      "title": "[Python] [sklearn] Text features as input to sklearn interface",
      "user": {
        "login": "mlisovyi",
        "id": 10157590,
        "node_id": "MDQ6VXNlcjEwMTU3NTkw",
        "avatar_url": "https://avatars0.githubusercontent.com/u/10157590?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/mlisovyi",
        "html_url": "https://github.com/mlisovyi",
        "followers_url": "https://api.github.com/users/mlisovyi/followers",
        "following_url": "https://api.github.com/users/mlisovyi/following{/other_user}",
        "gists_url": "https://api.github.com/users/mlisovyi/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/mlisovyi/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/mlisovyi/subscriptions",
        "organizations_url": "https://api.github.com/users/mlisovyi/orgs",
        "repos_url": "https://api.github.com/users/mlisovyi/repos",
        "events_url": "https://api.github.com/users/mlisovyi/events{/privacy}",
        "received_events_url": "https://api.github.com/users/mlisovyi/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 0,
      "created_at": "2018-04-30T18:58:56Z",
      "updated_at": "2018-04-30T18:58:56Z",
      "closed_at": null,
      "author_association": "CONTRIBUTOR",
      "body": "Hi all,\r\n\r\nIs there an example on how to use **text features in the VW sklearn API**? I can not make the `sklearn_vw.tovw()` function work. I'm interested in this function because tovw is used by sklearn_vw.VW to convert the regular input into VW-compatible format. There seems to be 2 separate problems, both related to `sklearn.datasets.svmlight_format.dump_svmlight_file()` usage by `tovw`. The first is a small feature, which can be worked around by additional type casting, but the second is unclear to me.\r\n\r\n1). There is a mis-handling of pandas.Series objects as an input, which seems to be due to the dtype being `'object'`. Below is a dummy example to reproduce the problem:\r\n```python\r\nimport pandas as pd\r\nimport numpy as np\r\nfrom vowpalwabbit.sklearn_vw import tovw\r\n\r\nX = pd.Series(['cat', 'dog', 'cat', 'cat'], name='catdog')\r\ny = pd.Series([-1, 1, -1, -1], name='label')\r\n\r\ntovw(X, y)\r\n```\r\nThe output is \r\n```python\r\n---------------------------------------------------------------------------\r\nValueError                                Traceback (most recent call last)\r\n<ipython-input-4-705a017a4383> in <module>()\r\n      8 y = pd.Series([-1, 1, -1, -1], name='label')\r\n      9 \r\n---> 10 tovw(X, y)\r\n\r\n.../vowpalwabbit/sklearn_vw.py in tovw(x, y, sample_weight)\r\n    579     # convert input to svmlight format\r\n    580     s = io.BytesIO()\r\n--> 581     dump_svmlight_file(x, np.zeros(rows), s)\r\n    582 \r\n    583     # parse entries to construct VW format\r\n\r\n.../sklearn/datasets/svmlight_format.py in dump_svmlight_file(X, y, f, zero_based, comment, query_id, multilabel)\r\n    445                              % (yval.shape,))\r\n    446 \r\n--> 447     Xval = check_array(X, accept_sparse='csr')\r\n    448     if Xval.shape[0] != yval.shape[0]:\r\n    449         raise ValueError(\"X.shape[0] and y.shape[0] should be the same, got\"\r\n\r\n.../sklearn/utils/validation.py in check_array(array, accept_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, ensure_min_samples, ensure_min_features, warn_on_dtype, estimator)\r\n    431                                       force_all_finite)\r\n    432     else:\r\n--> 433         array = np.array(array, dtype=dtype, order=order, copy=copy)\r\n    434 \r\n    435         if ensure_2d:\r\n\r\nValueError: could not convert string to float: 'cat'\r\n```\r\nThis can be resolved by passing a `np.array` or strings, i.e. calling `tovw(X.values.astype(str), y)`. This is not elegant and multiple users will bump into the same problem, but it is resolvable. However, this brings us to the second problem.\r\n\r\n2). Giving an array of string as X input still leads to another issue with `sklearn.datasets.svmlight_format.dump_svmlight_file()`:\r\n\r\n```python\r\n---------------------------------------------------------------------------\r\nTypeError                                 Traceback (most recent call last)\r\n<ipython-input-11-42a921e37bc9> in <module>()\r\n      2 y_a = np.array([-1, 1, -1, -1])\r\n      3 \r\n----> 4 tovw(X_a, y_a)\r\n\r\n.../vowpalwabbit/sklearn_vw.py in tovw(x, y, sample_weight)\r\n    579     # convert input to svmlight format\r\n    580     s = io.BytesIO()\r\n--> 581     dump_svmlight_file(x, np.zeros(rows), s)\r\n    582 \r\n    583     # parse entries to construct VW format\r\n\r\n.../sklearn/datasets/svmlight_format.py in dump_svmlight_file(X, y, f, zero_based, comment, query_id, multilabel)\r\n    476 \r\n    477     if hasattr(f, \"write\"):\r\n--> 478         _dump_svmlight(X, y, f, multilabel, one_based, comment, query_id)\r\n    479     else:\r\n    480         with open(f, \"wb\") as f:\r\n.../sklearn/datasets/svmlight_format.py in _dump_svmlight(X, y, f, multilabel, one_based, comment, query_id)\r\n    352             row = zip(np.where(nz)[0], X[i, nz])\r\n    353 \r\n--> 354         s = \" \".join(value_pattern % (j + one_based, x) for j, x in row)\r\n    355 \r\n    356         if multilabel:\r\n.../sklearn/datasets/svmlight_format.py in <genexpr>(.0)\r\n    352             row = zip(np.where(nz)[0], X[i, nz])\r\n    353 \r\n--> 354         s = \" \".join(value_pattern % (j + one_based, x) for j, x in row)\r\n    355 \r\n    356         if multilabel:\r\n\r\nTypeError: only size-1 arrays can be converted to Python scalars\r\n```\r\n\r\n**Environmental setup**\r\nUbuntu 16.04 with conda 4.5.1 and python3.5 \r\nvowpalwabbit  8.5.0 (installed from pypi), scikit-learn 0.19.1, numpy  1.14.2\r\n\r\nAny feedback/help would be appreciated.\r\n\r\nCheers,\r\nMisha",
      "score": 94.51168
    },
    {
      "url": "https://api.github.com/repos/JohnLangford/vowpal_wabbit/issues/1475",
      "repository_url": "https://api.github.com/repos/JohnLangford/vowpal_wabbit",
      "labels_url": "https://api.github.com/repos/JohnLangford/vowpal_wabbit/issues/1475/labels{/name}",
      "comments_url": "https://api.github.com/repos/JohnLangford/vowpal_wabbit/issues/1475/comments",
      "events_url": "https://api.github.com/repos/JohnLangford/vowpal_wabbit/issues/1475/events",
      "html_url": "https://github.com/JohnLangford/vowpal_wabbit/issues/1475",
      "id": 319455480,
      "node_id": "MDU6SXNzdWUzMTk0NTU0ODA=",
      "number": 1475,
      "title": "[Python] [sklearn] A few issues (and minor bugs?) in the sklearn_vw.VW.fit() method",
      "user": {
        "login": "mlisovyi",
        "id": 10157590,
        "node_id": "MDQ6VXNlcjEwMTU3NTkw",
        "avatar_url": "https://avatars0.githubusercontent.com/u/10157590?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/mlisovyi",
        "html_url": "https://github.com/mlisovyi",
        "followers_url": "https://api.github.com/users/mlisovyi/followers",
        "following_url": "https://api.github.com/users/mlisovyi/following{/other_user}",
        "gists_url": "https://api.github.com/users/mlisovyi/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/mlisovyi/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/mlisovyi/subscriptions",
        "organizations_url": "https://api.github.com/users/mlisovyi/orgs",
        "repos_url": "https://api.github.com/users/mlisovyi/repos",
        "events_url": "https://api.github.com/users/mlisovyi/events{/privacy}",
        "received_events_url": "https://api.github.com/users/mlisovyi/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 1,
      "created_at": "2018-05-02T08:21:05Z",
      "updated_at": "2018-06-08T11:23:28Z",
      "closed_at": null,
      "author_association": "CONTRIBUTOR",
      "body": "Hi,\r\n\r\nplaying around with the sklearn interface of VW it seems there a few issues (applies to v 8.5.0 and the current master branch):\r\n1. A **very minor bug**: it seems that the code intends to shuffle examples between epochs/passes. This is done [here in the code](https://github.com/JohnLangford/vowpal_wabbit/blob/master/python/vowpalwabbit/sklearn_vw.py#L307). However, `range(n)` starts with `0`, so the `n > 1` condition has to be changed to `n >= 1` otherwise the second pass is done without reshuffling.\r\n2. An **issue/bug** (?): the current implementation of shuffling doesn't work in a general case for pd.DataFrame or pd.Series as an input. It will work if all indices are sequential starting from 0. However, if one attempts to use a KFold splitting from sklearn, then it breaks down. A minimal example to reproduce the problem is given at the end of the post. This feature can be worked around by passing `pd.Series.values` but at the moment it is not clear from the documentation that pandas objects directly have only a limited support. Pandas support also relates to observations in #1472.\r\n3. An **issue**: even if a `np.array` is passed directly, the `VW.fit()` with several passes shuffles examples in the original array. The problem with this is that one wants to evaluate performance on tthe training dataset, thus doing `clf.predict()` and to compare to a vector (=array/pd.Series/etc.) true labels. True labels are usually available separately in a sklearn nomenclature (i.e. 0/1 labels instead of -1/1). However, those have not been shuffled and thus the order of predictions and true labels is broken and performance metric evaluation doesn't make sense. Is it possible to create a local copy of the input data object inside the `.fit()` method and clear it at the end? Otherwise, the documentation has to be very explicit about this behaviour.\r\n\r\nThanks in advance for feedback\r\n\r\n**Environmental setup**\r\nUbuntu 16.04 with conda 4.5.1 and python3.5\r\nvowpalwabbit 8.5.0 (installed from pypi), scikit-learn 0.19.1, numpy 1.14.2\r\n\r\n**Example of the issue 2**\r\n```python\r\nimport pandas as pd\r\nimport numpy as np\r\nfrom vowpalwabbit.sklearn_vw import VWClassifier\r\nfrom sklearn.model_selection import KFold\r\n\r\n# The dummy example X has already been adjusted to the native VW format, so convert_to_vw=False\r\nX = pd.Series(['1 |Pet cat', '-1 |Pet dog', '1 |Pet cat', '1 |Pet cat'], name='catdog')\r\n# The sklearn-friendly y is needed for the KFold.split() only\r\ny = pd.Series([-1, 1, -1, -1], name='label')\r\n\r\nkfold = KFold(n_splits=3, random_state=314, shuffle=False)\r\n\r\n# Split the data in folds for cross validation\r\nfor train_idx, valid_idx in kfold.split(X, y):\r\n    train_cv, valid_cv = X[train_idx], X[valid_idx]\r\n    # Classifier with a single pass over the data \r\n    clf = VWClassifier(passes=1, convert_to_vw=False)\r\n    # This will work\r\n    clf.fit(train_cv)\r\n    print('done with a single pass')\r\n\r\n    # Classifier with a 3 passes over the data  (i.e. reshuffling should take place)\r\n    clf = VWClassifier(passes=3, convert_to_vw=False)\r\n    # This will still work\r\n    clf.fit(train_cv.values)\r\n    print('done with 3 passes on np.array')\r\n    # This will NOT work\r\n    print('Current indices: ' + str(train_cv.index.values))\r\n    clf.fit(train_cv)\r\n    print('done with 3 passes on pd.Series')\r\n```\r\nThe output is:\r\n```python \r\ndone with a single pass\r\ndone with 3 passes on np.array\r\nCurrent indices: [2 3]\r\n\r\n---------------------------------------------------------------------------\r\nKeyError                                  Traceback (most recent call last)\r\n<ipython-input-12-43a4b42f3a13> in <module>()\r\n     26     # This will NOT work\r\n     27     print('Current indices: ' + str(train_cv.index.values))\r\n---> 28     clf.fit(train_cv)\r\n     29     print('done with 3 passes on pd.Series')\r\n\r\n~/anaconda2/envs/Titanic/lib/python3.5/site-packages/vowpalwabbit/sklearn_vw.py in fit(self, X, y, sample_weight)\r\n    306         for n in range(self.passes_):\r\n    307             if n > 1:\r\n--> 308                 np.random.shuffle(X)\r\n    309             for idx, x in enumerate(X):\r\n    310                 model.learn(x)\r\n\r\nmtrand.pyx in mtrand.RandomState.shuffle()\r\n\r\nmtrand.pyx in mtrand.RandomState.shuffle()\r\n\r\n~/anaconda2/envs/Titanic/lib/python3.5/site-packages/pandas/core/series.py in __getitem__(self, key)\r\n    621         key = com._apply_if_callable(key, self)\r\n    622         try:\r\n--> 623             result = self.index.get_value(self, key)\r\n    624 \r\n    625             if not is_scalar(result):\r\n\r\n~/anaconda2/envs/Titanic/lib/python3.5/site-packages/pandas/core/indexes/base.py in get_value(self, series, key)\r\n   2558         try:\r\n   2559             return self._engine.get_value(s, k,\r\n-> 2560                                           tz=getattr(series.dtype, 'tz', None))\r\n   2561         except KeyError as e1:\r\n   2562             if len(self) > 0 and self.inferred_type in ['integer', 'boolean']:\r\n\r\npandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_value()\r\n\r\npandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_value()\r\n\r\npandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()\r\n\r\npandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.Int64HashTable.get_item()\r\n\r\npandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.Int64HashTable.get_item()\r\n\r\nKeyError: 0\r\n\r\n```",
      "score": 85.419495
    },
    {
      "url": "https://api.github.com/repos/sbinet/go-trackml/issues/3",
      "repository_url": "https://api.github.com/repos/sbinet/go-trackml",
      "labels_url": "https://api.github.com/repos/sbinet/go-trackml/issues/3/labels{/name}",
      "comments_url": "https://api.github.com/repos/sbinet/go-trackml/issues/3/comments",
      "events_url": "https://api.github.com/repos/sbinet/go-trackml/issues/3/events",
      "html_url": "https://github.com/sbinet/go-trackml/issues/3",
      "id": 323524697,
      "node_id": "MDU6SXNzdWUzMjM1MjQ2OTc=",
      "number": 3,
      "title": "trackml: leverage go-sklearn to implement kNN",
      "user": {
        "login": "sbinet",
        "id": 1070920,
        "node_id": "MDQ6VXNlcjEwNzA5MjA=",
        "avatar_url": "https://avatars2.githubusercontent.com/u/1070920?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/sbinet",
        "html_url": "https://github.com/sbinet",
        "followers_url": "https://api.github.com/users/sbinet/followers",
        "following_url": "https://api.github.com/users/sbinet/following{/other_user}",
        "gists_url": "https://api.github.com/users/sbinet/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/sbinet/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/sbinet/subscriptions",
        "organizations_url": "https://api.github.com/users/sbinet/orgs",
        "repos_url": "https://api.github.com/users/sbinet/repos",
        "events_url": "https://api.github.com/users/sbinet/events{/privacy}",
        "received_events_url": "https://api.github.com/users/sbinet/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 0,
      "created_at": "2018-05-16T08:48:27Z",
      "updated_at": "2018-05-16T08:48:27Z",
      "closed_at": null,
      "author_association": "OWNER",
      "body": "consider using [go-sklearn](https://godoc.org/github.com/pa-m/sklearn) to implement the other python starter kit example (kNN.ipynb):\r\n\r\n```python\r\nfrom sklearn.preprocessing import StandardScaler\r\nfrom sklearn.neighbors import KNeighborsClassifier\r\n```",
      "score": 81.63792
    },
    {
      "url": "https://api.github.com/repos/sbinet/go-trackml/issues/2",
      "repository_url": "https://api.github.com/repos/sbinet/go-trackml",
      "labels_url": "https://api.github.com/repos/sbinet/go-trackml/issues/2/labels{/name}",
      "comments_url": "https://api.github.com/repos/sbinet/go-trackml/issues/2/comments",
      "events_url": "https://api.github.com/repos/sbinet/go-trackml/issues/2/events",
      "html_url": "https://github.com/sbinet/go-trackml/issues/2",
      "id": 323524295,
      "node_id": "MDU6SXNzdWUzMjM1MjQyOTU=",
      "number": 2,
      "title": "trackml: leverage go-sklearn to implement DBSCAN",
      "user": {
        "login": "sbinet",
        "id": 1070920,
        "node_id": "MDQ6VXNlcjEwNzA5MjA=",
        "avatar_url": "https://avatars2.githubusercontent.com/u/1070920?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/sbinet",
        "html_url": "https://github.com/sbinet",
        "followers_url": "https://api.github.com/users/sbinet/followers",
        "following_url": "https://api.github.com/users/sbinet/following{/other_user}",
        "gists_url": "https://api.github.com/users/sbinet/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/sbinet/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/sbinet/subscriptions",
        "organizations_url": "https://api.github.com/users/sbinet/orgs",
        "repos_url": "https://api.github.com/users/sbinet/repos",
        "events_url": "https://api.github.com/users/sbinet/events{/privacy}",
        "received_events_url": "https://api.github.com/users/sbinet/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 0,
      "created_at": "2018-05-16T08:47:19Z",
      "updated_at": "2018-05-16T08:48:47Z",
      "closed_at": null,
      "author_association": "OWNER",
      "body": "consider using [go-sklearn](https://godoc.org/github.com/pa-m/sklearn) to implement the other python starter kit example (DBSCAN.ipynb):\r\n\r\n```python\r\nfrom sklearn.preprocessing import StandardScaler\r\nfrom sklearn.cluster import DBSCAN\r\n\r\nclass Clusterer(object):\r\n   ## blabla\r\n   pass\r\n```",
      "score": 78.814804
    },
    {
      "url": "https://api.github.com/repos/automl/auto-sklearn/issues/488",
      "repository_url": "https://api.github.com/repos/automl/auto-sklearn",
      "labels_url": "https://api.github.com/repos/automl/auto-sklearn/issues/488/labels{/name}",
      "comments_url": "https://api.github.com/repos/automl/auto-sklearn/issues/488/comments",
      "events_url": "https://api.github.com/repos/automl/auto-sklearn/issues/488/events",
      "html_url": "https://github.com/automl/auto-sklearn/issues/488",
      "id": 331224031,
      "node_id": "MDU6SXNzdWUzMzEyMjQwMzE=",
      "number": 488,
      "title": "Error while installing auto-sklearn - \"failed building wheel for pyrfr\"",
      "user": {
        "login": "sundarsri96",
        "id": 38464895,
        "node_id": "MDQ6VXNlcjM4NDY0ODk1",
        "avatar_url": "https://avatars0.githubusercontent.com/u/38464895?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/sundarsri96",
        "html_url": "https://github.com/sundarsri96",
        "followers_url": "https://api.github.com/users/sundarsri96/followers",
        "following_url": "https://api.github.com/users/sundarsri96/following{/other_user}",
        "gists_url": "https://api.github.com/users/sundarsri96/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/sundarsri96/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/sundarsri96/subscriptions",
        "organizations_url": "https://api.github.com/users/sundarsri96/orgs",
        "repos_url": "https://api.github.com/users/sundarsri96/repos",
        "events_url": "https://api.github.com/users/sundarsri96/events{/privacy}",
        "received_events_url": "https://api.github.com/users/sundarsri96/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 3,
      "created_at": "2018-06-11T14:54:01Z",
      "updated_at": "2018-06-12T16:16:22Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "The installation for auto-sklearn is not working because it throws the error that \"failed building wheel for pyrfr\". Can you help me out with this, please?\r\n\r\nInput:\r\npip install auto-sklearn\r\n\r\nOutput:\r\nCollecting auto-sklearn\r\nRequirement already satisfied: joblib in ./anaconda3/lib/python3.6/site-packages (from auto-sklearn) (0.11)\r\nRequirement already satisfied: ConfigSpace<0.5,>=0.4.0 in ./anaconda3/lib/python3.6/site-packages (from auto-sklearn) (0.4.6)\r\nRequirement already satisfied: lockfile in ./anaconda3/lib/python3.6/site-packages (from auto-sklearn) (0.12.2)\r\nRequirement already satisfied: pynisher<0.5,>=0.4 in ./anaconda3/lib/python3.6/site-packages (from auto-sklearn) (0.4.2)\r\nCollecting smac<0.9,>=0.8 (from auto-sklearn)\r\nRequirement already satisfied: setuptools in ./anaconda3/lib/python3.6/site-packages (from auto-sklearn) (39.1.0)\r\nRequirement already satisfied: six in ./anaconda3/lib/python3.6/site-packages (from auto-sklearn) (1.11.0)\r\nRequirement already satisfied: Cython in ./anaconda3/lib/python3.6/site-packages (from auto-sklearn) (0.28.2)\r\nRequirement already satisfied: nose in ./anaconda3/lib/python3.6/site-packages (from auto-sklearn) (1.3.7)\r\nRequirement already satisfied: scikit-learn<0.20,>=0.19 in ./anaconda3/lib/python3.6/site-packages (from auto-sklearn) (0.19.1)\r\nCollecting pyrfr<0.8,>=0.6.1 (from auto-sklearn)\r\n  Using cached https://files.pythonhosted.org/packages/c3/c6/c555cfa3c7d0078dded091d4901ed52344bbb925077aa70b871faf35fd58/pyrfr-0.7.4.tar.gz\r\nRequirement already satisfied: pandas in ./anaconda3/lib/python3.6/site-packages (from auto-sklearn) (0.23.0)\r\nRequirement already satisfied: psutil in ./anaconda3/lib/python3.6/site-packages (from auto-sklearn) (5.4.5)\r\nRequirement already satisfied: pyyaml in ./anaconda3/lib/python3.6/site-packages (from auto-sklearn) (3.12)\r\nRequirement already satisfied: numpy>=1.9.0 in ./anaconda3/lib/python3.6/site-packages (from auto-sklearn) (1.14.3)\r\nRequirement already satisfied: scipy>=0.14.1 in ./anaconda3/lib/python3.6/site-packages (from auto-sklearn) (1.1.0)\r\nRequirement already satisfied: liac-arff in ./anaconda3/lib/python3.6/site-packages (from auto-sklearn) (2.2.1)\r\nRequirement already satisfied: typing in ./anaconda3/lib/python3.6/site-packages (from ConfigSpace<0.5,>=0.4.0->auto-sklearn) (3.6.4)\r\nRequirement already satisfied: pyparsing in ./anaconda3/lib/python3.6/site-packages (from ConfigSpace<0.5,>=0.4.0->auto-sklearn) (2.2.0)\r\nRequirement already satisfied: docutils>=0.3 in ./anaconda3/lib/python3.6/site-packages (from pynisher<0.5,>=0.4->auto-sklearn) (0.14)\r\nRequirement already satisfied: sphinx in ./anaconda3/lib/python3.6/site-packages (from smac<0.9,>=0.8->auto-sklearn) (1.7.4)\r\nCollecting sphinx-rtd-theme (from smac<0.9,>=0.8->auto-sklearn)\r\n  Using cached https://files.pythonhosted.org/packages/5d/01/f367780d51e9b353cec890257c84202a340f645d5a037b61a52c93669764/sphinx_rtd_theme-0.4.0-py2.py3-none-any.whl\r\nRequirement already satisfied: python-dateutil>=2.5.0 in ./anaconda3/lib/python3.6/site-packages (from pandas->auto-sklearn) (2.7.3)\r\nRequirement already satisfied: pytz>=2011k in ./anaconda3/lib/python3.6/site-packages (from pandas->auto-sklearn) (2018.4)\r\nRequirement already satisfied: Jinja2>=2.3 in ./anaconda3/lib/python3.6/site-packages (from sphinx->smac<0.9,>=0.8->auto-sklearn) (2.10)\r\nRequirement already satisfied: Pygments>=2.0 in ./anaconda3/lib/python3.6/site-packages (from sphinx->smac<0.9,>=0.8->auto-sklearn) (2.2.0)\r\nRequirement already satisfied: snowballstemmer>=1.1 in ./anaconda3/lib/python3.6/site-packages (from sphinx->smac<0.9,>=0.8->auto-sklearn) (1.2.1)\r\nRequirement already satisfied: babel!=2.0,>=1.3 in ./anaconda3/lib/python3.6/site-packages (from sphinx->smac<0.9,>=0.8->auto-sklearn) (2.5.3)\r\nRequirement already satisfied: alabaster<0.8,>=0.7 in ./anaconda3/lib/python3.6/site-packages (from sphinx->smac<0.9,>=0.8->auto-sklearn) (0.7.10)\r\nRequirement already satisfied: imagesize in ./anaconda3/lib/python3.6/site-packages (from sphinx->smac<0.9,>=0.8->auto-sklearn) (1.0.0)\r\nRequirement already satisfied: requests>=2.0.0 in ./anaconda3/lib/python3.6/site-packages (from sphinx->smac<0.9,>=0.8->auto-sklearn) (2.18.4)\r\nRequirement already satisfied: packaging in ./anaconda3/lib/python3.6/site-packages (from sphinx->smac<0.9,>=0.8->auto-sklearn) (17.1)\r\nRequirement already satisfied: sphinxcontrib-websupport in ./anaconda3/lib/python3.6/site-packages (from sphinx->smac<0.9,>=0.8->auto-sklearn) (1.0.1)\r\nRequirement already satisfied: MarkupSafe>=0.23 in ./anaconda3/lib/python3.6/site-packages (from Jinja2>=2.3->sphinx->smac<0.9,>=0.8->auto-sklearn) (1.0)\r\nRequirement already satisfied: chardet<3.1.0,>=3.0.2 in ./anaconda3/lib/python3.6/site-packages (from requests>=2.0.0->sphinx->smac<0.9,>=0.8->auto-sklearn) (3.0.4)\r\nRequirement already satisfied: idna<2.7,>=2.5 in ./anaconda3/lib/python3.6/site-packages (from requests>=2.0.0->sphinx->smac<0.9,>=0.8->auto-sklearn) (2.6)\r\nRequirement already satisfied: urllib3<1.23,>=1.21.1 in ./anaconda3/lib/python3.6/site-packages (from requests>=2.0.0->sphinx->smac<0.9,>=0.8->auto-sklearn) (1.22)\r\nRequirement already satisfied: certifi>=2017.4.17 in ./anaconda3/lib/python3.6/site-packages (from requests>=2.0.0->sphinx->smac<0.9,>=0.8->auto-sklearn) (2018.4.16)\r\nBuilding wheels for collected packages: pyrfr\r\n  Running setup.py bdist_wheel for pyrfr ... error\r\n  Complete output from command /home/aifordesign/anaconda3/bin/python -u -c \"import setuptools, tokenize;__file__='/tmp/pip-install-bhn5nmx1/pyrfr/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\\r\\n', '\\n');f.close();exec(compile(code, __file__, 'exec'))\" bdist_wheel -d /tmp/pip-wheel-0l_njh38 --python-tag cp36:\r\n  running bdist_wheel\r\n  running build\r\n  running build_ext\r\n  building 'pyrfr._regression' extension\r\n  swigging pyrfr/regression.i to pyrfr/regression_wrap.cpp\r\n  swig -python -c++ -modern -features nondynamic -I./include -o pyrfr/regression_wrap.cpp pyrfr/regression.i\r\n  unable to execute 'swig': No such file or directory\r\n  error: command 'swig' failed with exit status 1\r\n  \r\n  ----------------------------------------\r\n  Failed building wheel for pyrfr\r\n  Running setup.py clean for pyrfr\r\nFailed to build pyrfr\r\ndistributed 1.21.8 requires msgpack, which is not installed.\r\ntensorboard 1.7.0 has requirement bleach==1.5.0, but you'll have bleach 2.1.3 which is incompatible.\r\nbleach 2.1.3 has requirement html5lib!=1.0b1,!=1.0b2,!=1.0b3,!=1.0b4,!=1.0b5,!=1.0b6,!=1.0b7,!=1.0b8,>=0.99999999pre, but you'll have html5lib 0.9999999 which is incompatible.\r\nInstalling collected packages: pyrfr, sphinx-rtd-theme, smac, auto-sklearn\r\n  Running setup.py install for pyrfr ... error\r\n    Complete output from command /home/aifordesign/anaconda3/bin/python -u -c \"import setuptools, tokenize;__file__='/tmp/pip-install-bhn5nmx1/pyrfr/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\\r\\n', '\\n');f.close();exec(compile(code, __file__, 'exec'))\" install --record /tmp/pip-record-dmn3u4d8/install-record.txt --single-version-externally-managed --compile:\r\n    running install\r\n    running build_ext\r\n    building 'pyrfr._regression' extension\r\n    swigging pyrfr/regression.i to pyrfr/regression_wrap.cpp\r\n    swig -python -c++ -modern -features nondynamic -I./include -o pyrfr/regression_wrap.cpp pyrfr/regression.i\r\n    unable to execute 'swig': No such file or directory\r\n    error: command 'swig' failed with exit status 1\r\n    \r\n    ----------------------------------------\r\nCommand \"/home/aifordesign/anaconda3/bin/python -u -c \"import setuptools, tokenize;__file__='/tmp/pip-install-bhn5nmx1/pyrfr/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\\r\\n', '\\n');f.close();exec(compile(code, __file__, 'exec'))\" install --record /tmp/pip-record-dmn3u4d8/install-record.txt --single-version-externally-managed --compile\" failed with error code 1 in /tmp/pip-install-bhn5nmx1/pyrfr/\r\n",
      "score": 77.06703
    },
    {
      "url": "https://api.github.com/repos/JohnLangford/vowpal_wabbit/issues/1469",
      "repository_url": "https://api.github.com/repos/JohnLangford/vowpal_wabbit",
      "labels_url": "https://api.github.com/repos/JohnLangford/vowpal_wabbit/issues/1469/labels{/name}",
      "comments_url": "https://api.github.com/repos/JohnLangford/vowpal_wabbit/issues/1469/comments",
      "events_url": "https://api.github.com/repos/JohnLangford/vowpal_wabbit/issues/1469/events",
      "html_url": "https://github.com/JohnLangford/vowpal_wabbit/issues/1469",
      "id": 318353946,
      "node_id": "MDU6SXNzdWUzMTgzNTM5NDY=",
      "number": 1469,
      "title": "[Python] [sklearn] Documentation of the API and a few questions",
      "user": {
        "login": "mlisovyi",
        "id": 10157590,
        "node_id": "MDQ6VXNlcjEwMTU3NTkw",
        "avatar_url": "https://avatars0.githubusercontent.com/u/10157590?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/mlisovyi",
        "html_url": "https://github.com/mlisovyi",
        "followers_url": "https://api.github.com/users/mlisovyi/followers",
        "following_url": "https://api.github.com/users/mlisovyi/following{/other_user}",
        "gists_url": "https://api.github.com/users/mlisovyi/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/mlisovyi/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/mlisovyi/subscriptions",
        "organizations_url": "https://api.github.com/users/mlisovyi/orgs",
        "repos_url": "https://api.github.com/users/mlisovyi/repos",
        "events_url": "https://api.github.com/users/mlisovyi/events{/privacy}",
        "received_events_url": "https://api.github.com/users/mlisovyi/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 0,
      "created_at": "2018-04-27T10:14:35Z",
      "updated_at": "2018-04-27T10:14:35Z",
      "closed_at": null,
      "author_association": "CONTRIBUTOR",
      "body": "Hi,\r\n\r\nI'm just starting to use VW in python. I'm very happy that there is a python binding and one dores not need to go around to dump `.vw` files with custom functions, running from a command line and then reading in the predictions in python again. And in particular, the sklearn wrapper is very useful, as it allows to use multiple handy functionalities from that package and makes usage universal across many algorithms. So thanks a lot!\r\n\r\nI have a few questions to this interface:\r\n1. Is there a documentation of the API (pyvw in general, and sklearn_vw in particular)? Would it be possible to add auto-generated docs from the class and function description? For example, one can use [Sphinx](http://www.sphinx-doc.org/en/master/) for that or any other tool that you would fancy. This will help users to find information faster.\r\n2. I wanted to understand how exactly training is handled by the sklearn interface. It seems that there are 2 regimes: \r\n2.1 VW native approach based on a `.vw` file. This seems to be enabled by the `data` arg in the constructor and learning done in the `pyvw.vw()` instantiation in `get_vw()`. Is that understanding correct? If so, this is very nice as it allows online training with a restricted memory footprint, as advertised by the tool. However, from [here](https://github.com/JohnLangford/vowpal_wabbit/blob/master/python/vowpalwabbit/sklearn_vw.py#L251), it seems that even in this case one runs one (and always only one) _additional_ pass over the data provided to the `.fit()` function. Would it make sense to set the number of passes to 0 in this case?\r\n2.2 The sklearn-like interface based on in-memory X and y arrays. In this case one iterates over `pyvw.vw.learn()` calls. That's fine for small datasets, I imagine. It also makes start of VW usage for people with sklearn experience much faster and smoother. The only thing here, would it be possible to add an additional parameter to the constructor that would be specific to the sklearn API and would allow one to convert `[0,1]` labels in the classification into `[-1,1]`. This simplifies life a lot, since in most cases data have `0` label instead of `-1`. One can maybe even reuse the existing `convert_to_vw_` attribute to trigger this conversion, but this would not be backward-compatible (I'm not sure how many people use the sklearn interface of VW by now).\r\n\r\nRelated to the approach discussed in 2.1, why is `fit_` attritube set to `True` already in the constructor in this case? It seems that the actual fit is done still inside `get_vw()` call from `.fit()`. I might have got it wrong, but if I'm right, than this defies the definition of the `fit_` attribute, as it leads to actual model fit inside `.predict()` call, which would make this call slow and confuse a user, is `.fit()` was never called.\r\n\r\nI'm happy to help with implementation of some of the issues discussed above, if you find them useful.\r\n\r\nCheers,\r\nMisha",
      "score": 73.60126
    },
    {
      "url": "https://api.github.com/repos/rsteca/sklearn-deap/issues/52",
      "repository_url": "https://api.github.com/repos/rsteca/sklearn-deap",
      "labels_url": "https://api.github.com/repos/rsteca/sklearn-deap/issues/52/labels{/name}",
      "comments_url": "https://api.github.com/repos/rsteca/sklearn-deap/issues/52/comments",
      "events_url": "https://api.github.com/repos/rsteca/sklearn-deap/issues/52/events",
      "html_url": "https://github.com/rsteca/sklearn-deap/issues/52",
      "id": 330367591,
      "node_id": "MDU6SXNzdWUzMzAzNjc1OTE=",
      "number": 52,
      "title": "Running Search with n_jobs>1 throws Deap Attribute Error. Official sklearn example",
      "user": {
        "login": "luded",
        "id": 36469747,
        "node_id": "MDQ6VXNlcjM2NDY5NzQ3",
        "avatar_url": "https://avatars0.githubusercontent.com/u/36469747?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/luded",
        "html_url": "https://github.com/luded",
        "followers_url": "https://api.github.com/users/luded/followers",
        "following_url": "https://api.github.com/users/luded/following{/other_user}",
        "gists_url": "https://api.github.com/users/luded/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/luded/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/luded/subscriptions",
        "organizations_url": "https://api.github.com/users/luded/orgs",
        "repos_url": "https://api.github.com/users/luded/repos",
        "events_url": "https://api.github.com/users/luded/events{/privacy}",
        "received_events_url": "https://api.github.com/users/luded/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 2,
      "created_at": "2018-06-07T17:31:08Z",
      "updated_at": "2018-06-08T16:24:00Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "code downloaded from sklearn for GridSearchCV\r\nreplaced by EvolutionaryAlgorithmSearchCV. Works only for n_jobs=1. Else throws a Deap exception.\r\n\r\npython code attached for reproduction.\r\n\r\n[grid_search_text_feature_extraction.txt](https://github.com/rsteca/sklearn-deap/files/2081485/grid_search_text_feature_extraction.txt)\r\n",
      "score": 73.493324
    },
    {
      "url": "https://api.github.com/repos/kevingo/bookmarks/issues/186",
      "repository_url": "https://api.github.com/repos/kevingo/bookmarks",
      "labels_url": "https://api.github.com/repos/kevingo/bookmarks/issues/186/labels{/name}",
      "comments_url": "https://api.github.com/repos/kevingo/bookmarks/issues/186/comments",
      "events_url": "https://api.github.com/repos/kevingo/bookmarks/issues/186/events",
      "html_url": "https://github.com/kevingo/bookmarks/issues/186",
      "id": 304242215,
      "node_id": "MDU6SXNzdWUzMDQyNDIyMTU=",
      "number": 186,
      "title": "Naive Bayes Classifier with sklearn python example ",
      "user": {
        "login": "kevingo",
        "id": 1113954,
        "node_id": "MDQ6VXNlcjExMTM5NTQ=",
        "avatar_url": "https://avatars1.githubusercontent.com/u/1113954?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/kevingo",
        "html_url": "https://github.com/kevingo",
        "followers_url": "https://api.github.com/users/kevingo/followers",
        "following_url": "https://api.github.com/users/kevingo/following{/other_user}",
        "gists_url": "https://api.github.com/users/kevingo/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/kevingo/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/kevingo/subscriptions",
        "organizations_url": "https://api.github.com/users/kevingo/orgs",
        "repos_url": "https://api.github.com/users/kevingo/repos",
        "events_url": "https://api.github.com/users/kevingo/events{/privacy}",
        "received_events_url": "https://api.github.com/users/kevingo/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [
        {
          "id": 624335372,
          "node_id": "MDU6TGFiZWw2MjQzMzUzNzI=",
          "url": "https://api.github.com/repos/kevingo/bookmarks/labels/ML",
          "name": "ML",
          "color": "e99695",
          "default": false
        },
        {
          "id": 624335042,
          "node_id": "MDU6TGFiZWw2MjQzMzUwNDI=",
          "url": "https://api.github.com/repos/kevingo/bookmarks/labels/Python",
          "name": "Python",
          "color": "e99695",
          "default": false
        }
      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 0,
      "created_at": "2018-03-12T05:05:32Z",
      "updated_at": "2018-03-12T05:05:32Z",
      "closed_at": null,
      "author_association": "OWNER",
      "body": "http://twitter.com/kevingo/status/973061716517900288",
      "score": 73.17868
    },
    {
      "url": "https://api.github.com/repos/rickyi22/rickyi22.github.io/issues/1",
      "repository_url": "https://api.github.com/repos/rickyi22/rickyi22.github.io",
      "labels_url": "https://api.github.com/repos/rickyi22/rickyi22.github.io/issues/1/labels{/name}",
      "comments_url": "https://api.github.com/repos/rickyi22/rickyi22.github.io/issues/1/comments",
      "events_url": "https://api.github.com/repos/rickyi22/rickyi22.github.io/issues/1/events",
      "html_url": "https://github.com/rickyi22/rickyi22.github.io/issues/1",
      "id": 214683461,
      "node_id": "MDU6SXNzdWUyMTQ2ODM0NjE=",
      "number": 1,
      "title": "python sklearn",
      "user": {
        "login": "rickyi22",
        "id": 20011276,
        "node_id": "MDQ6VXNlcjIwMDExMjc2",
        "avatar_url": "https://avatars3.githubusercontent.com/u/20011276?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/rickyi22",
        "html_url": "https://github.com/rickyi22",
        "followers_url": "https://api.github.com/users/rickyi22/followers",
        "following_url": "https://api.github.com/users/rickyi22/following{/other_user}",
        "gists_url": "https://api.github.com/users/rickyi22/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/rickyi22/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/rickyi22/subscriptions",
        "organizations_url": "https://api.github.com/users/rickyi22/orgs",
        "repos_url": "https://api.github.com/users/rickyi22/repos",
        "events_url": "https://api.github.com/users/rickyi22/events{/privacy}",
        "received_events_url": "https://api.github.com/users/rickyi22/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 0,
      "created_at": "2017-03-16T12:07:06Z",
      "updated_at": "2017-03-16T12:07:06Z",
      "closed_at": null,
      "author_association": "OWNER",
      "body": "http://radimrehurek.com/data_science_python/",
      "score": 72.77731
    },
    {
      "url": "https://api.github.com/repos/JackYumCha/QA/issues/12",
      "repository_url": "https://api.github.com/repos/JackYumCha/QA",
      "labels_url": "https://api.github.com/repos/JackYumCha/QA/issues/12/labels{/name}",
      "comments_url": "https://api.github.com/repos/JackYumCha/QA/issues/12/comments",
      "events_url": "https://api.github.com/repos/JackYumCha/QA/issues/12/events",
      "html_url": "https://github.com/JackYumCha/QA/issues/12",
      "id": 326782565,
      "node_id": "MDU6SXNzdWUzMjY3ODI1NjU=",
      "number": 12,
      "title": "2018-05-27 Operations",
      "user": {
        "login": "JackYumCha",
        "id": 37918930,
        "node_id": "MDQ6VXNlcjM3OTE4OTMw",
        "avatar_url": "https://avatars2.githubusercontent.com/u/37918930?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/JackYumCha",
        "html_url": "https://github.com/JackYumCha",
        "followers_url": "https://api.github.com/users/JackYumCha/followers",
        "following_url": "https://api.github.com/users/JackYumCha/following{/other_user}",
        "gists_url": "https://api.github.com/users/JackYumCha/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/JackYumCha/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/JackYumCha/subscriptions",
        "organizations_url": "https://api.github.com/users/JackYumCha/orgs",
        "repos_url": "https://api.github.com/users/JackYumCha/repos",
        "events_url": "https://api.github.com/users/JackYumCha/events{/privacy}",
        "received_events_url": "https://api.github.com/users/JackYumCha/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 26,
      "created_at": "2018-05-27T02:47:13Z",
      "updated_at": "2018-05-27T08:22:22Z",
      "closed_at": null,
      "author_association": "OWNER",
      "body": "需要在tensorflow环境(python 3.5)当中安装sklearn和grpc\r\n```\r\nactivate tensorflow\r\nconda install scikit-learn\r\npython -m pip install --upgrade pip\r\npip install grpcio-tools\r\n```",
      "score": 72.69573
    },
    {
      "url": "https://api.github.com/repos/shimaokasonse/NFGEC/issues/5",
      "repository_url": "https://api.github.com/repos/shimaokasonse/NFGEC",
      "labels_url": "https://api.github.com/repos/shimaokasonse/NFGEC/issues/5/labels{/name}",
      "comments_url": "https://api.github.com/repos/shimaokasonse/NFGEC/issues/5/comments",
      "events_url": "https://api.github.com/repos/shimaokasonse/NFGEC/issues/5/events",
      "html_url": "https://github.com/shimaokasonse/NFGEC/issues/5",
      "id": 322587601,
      "node_id": "MDU6SXNzdWUzMjI1ODc2MDE=",
      "number": 5,
      "title": "ImportError: No module named sklearn.externals",
      "user": {
        "login": "loveJasmine",
        "id": 18051187,
        "node_id": "MDQ6VXNlcjE4MDUxMTg3",
        "avatar_url": "https://avatars2.githubusercontent.com/u/18051187?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/loveJasmine",
        "html_url": "https://github.com/loveJasmine",
        "followers_url": "https://api.github.com/users/loveJasmine/followers",
        "following_url": "https://api.github.com/users/loveJasmine/following{/other_user}",
        "gists_url": "https://api.github.com/users/loveJasmine/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/loveJasmine/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/loveJasmine/subscriptions",
        "organizations_url": "https://api.github.com/users/loveJasmine/orgs",
        "repos_url": "https://api.github.com/users/loveJasmine/repos",
        "events_url": "https://api.github.com/users/loveJasmine/events{/privacy}",
        "received_events_url": "https://api.github.com/users/loveJasmine/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 1,
      "created_at": "2018-05-13T11:55:41Z",
      "updated_at": "2018-05-13T12:19:53Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "(venv) mldl@mldlUB1604:~/ub16_prj/NFGEC$ python train.py figer attentive\r\nTraceback (most recent call last):\r\n  File \"train.py\", line 3, in <module>\r\n    from sklearn.externals import joblib\r\nImportError: No module named sklearn.externals\r\n",
      "score": 72.648415
    },
    {
      "url": "https://api.github.com/repos/SeldonIO/seldon-core/issues/164",
      "repository_url": "https://api.github.com/repos/SeldonIO/seldon-core",
      "labels_url": "https://api.github.com/repos/SeldonIO/seldon-core/issues/164/labels{/name}",
      "comments_url": "https://api.github.com/repos/SeldonIO/seldon-core/issues/164/comments",
      "events_url": "https://api.github.com/repos/SeldonIO/seldon-core/issues/164/events",
      "html_url": "https://github.com/SeldonIO/seldon-core/issues/164",
      "id": 331255680,
      "node_id": "MDU6SXNzdWUzMzEyNTU2ODA=",
      "number": 164,
      "title": "Docker image build error with sklearn_iris_docker example",
      "user": {
        "login": "koverholt",
        "id": 303236,
        "node_id": "MDQ6VXNlcjMwMzIzNg==",
        "avatar_url": "https://avatars0.githubusercontent.com/u/303236?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/koverholt",
        "html_url": "https://github.com/koverholt",
        "followers_url": "https://api.github.com/users/koverholt/followers",
        "following_url": "https://api.github.com/users/koverholt/following{/other_user}",
        "gists_url": "https://api.github.com/users/koverholt/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/koverholt/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/koverholt/subscriptions",
        "organizations_url": "https://api.github.com/users/koverholt/orgs",
        "repos_url": "https://api.github.com/users/koverholt/repos",
        "events_url": "https://api.github.com/users/koverholt/events{/privacy}",
        "received_events_url": "https://api.github.com/users/koverholt/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": {
        "login": "gsunner",
        "id": 8726172,
        "node_id": "MDQ6VXNlcjg3MjYxNzI=",
        "avatar_url": "https://avatars2.githubusercontent.com/u/8726172?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/gsunner",
        "html_url": "https://github.com/gsunner",
        "followers_url": "https://api.github.com/users/gsunner/followers",
        "following_url": "https://api.github.com/users/gsunner/following{/other_user}",
        "gists_url": "https://api.github.com/users/gsunner/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/gsunner/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/gsunner/subscriptions",
        "organizations_url": "https://api.github.com/users/gsunner/orgs",
        "repos_url": "https://api.github.com/users/gsunner/repos",
        "events_url": "https://api.github.com/users/gsunner/events{/privacy}",
        "received_events_url": "https://api.github.com/users/gsunner/received_events",
        "type": "User",
        "site_admin": false
      },
      "assignees": [
        {
          "login": "gsunner",
          "id": 8726172,
          "node_id": "MDQ6VXNlcjg3MjYxNzI=",
          "avatar_url": "https://avatars2.githubusercontent.com/u/8726172?v=4",
          "gravatar_id": "",
          "url": "https://api.github.com/users/gsunner",
          "html_url": "https://github.com/gsunner",
          "followers_url": "https://api.github.com/users/gsunner/followers",
          "following_url": "https://api.github.com/users/gsunner/following{/other_user}",
          "gists_url": "https://api.github.com/users/gsunner/gists{/gist_id}",
          "starred_url": "https://api.github.com/users/gsunner/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/gsunner/subscriptions",
          "organizations_url": "https://api.github.com/users/gsunner/orgs",
          "repos_url": "https://api.github.com/users/gsunner/repos",
          "events_url": "https://api.github.com/users/gsunner/events{/privacy}",
          "received_events_url": "https://api.github.com/users/gsunner/received_events",
          "type": "User",
          "site_admin": false
        }
      ],
      "milestone": null,
      "comments": 4,
      "created_at": "2018-06-11T16:14:03Z",
      "updated_at": "2018-06-13T17:38:36Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "I'm unable to build the Docker image for the `sklearn_iris_docker` example. It's failing at the `numpy` build stage.\r\n\r\nI tried upgrading `pip` and adding some dependencies with `apk` per https://stackoverflow.com/questions/33421965/installing-numpy-on-docker-alpine, but I had the same issue.\r\n\r\nStacktrace:\r\n\r\n```\r\nTraceback (most recent call last):\r\n      File \"<string>\", line 1, in <module>\r\n      File \"/tmp/pip-build-fUc8dq/pandas/setup.py\", line 631, in <module>\r\n        **setuptools_kwargs)\r\n      File \"/usr/lib/python2.7/distutils/core.py\", line 111, in setup\r\n        _setup_distribution = dist = klass(attrs)\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/dist.py\", line 317, in __init__\r\n        self.fetch_build_eggs(attrs['setup_requires'])\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/dist.py\", line 372, in fetch_build_eggs\r\n        replace_conflicting=True,\r\n      File \"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py\", line 846, in resolve\r\n        dist = best[req.key] = env.best_match(req, ws, installer)\r\n      File \"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py\", line 1118, in best_match\r\n        return self.obtain(req, installer)\r\n      File \"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py\", line 1130, in obtain\r\n        return installer(requirement)\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/dist.py\", line 440, in fetch_build_egg\r\n        return cmd.easy_install(req)\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py\", line 674, in easy_install\r\n        return self.install_item(spec, dist.location, tmpdir, deps)\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py\", line 700, in install_item\r\n        dists = self.install_eggs(spec, download, tmpdir)\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py\", line 881, in install_eggs\r\n        return self.build_and_install(setup_script, setup_base)\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py\", line 1120, in build_and_install\r\n        self.run_setup(setup_script, setup_base, args)\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py\", line 1108, in run_setup\r\n        raise DistutilsError(\"Setup script exited with %s\" % (v.args[0],))\r\n    distutils.errors.DistutilsError: Setup script exited with error: Command \"gcc -fno-strict-aliasing -Os -fomit-frame-pointer -g -DNDEBUG -Os -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x100000 -fPIC -Ibuild/src.linux-x86_64-2.7/numpy/core/src/npymath -Inumpy/core/include -Ibuild/src.linux-x86_64-2.7/numpy/core/include/numpy -Inumpy/core/src/private -Inumpy/core/src -Inumpy/core -Inumpy/core/src/npymath -Inumpy/core/src/multiarray -Inumpy/core/src/umath -Inumpy/core/src/npysort -I/usr/include/python2.7 -Ibuild/src.linux-x86_64-2.7/numpy/core/src/private -Ibuild/src.linux-x86_64-2.7/numpy/core/src/npymath -Ibuild/src.linux-x86_64-2.7/numpy/core/src/private -Ibuild/src.linux-x86_64-2.7/numpy/core/src/npymath -Ibuild/src.linux-x86_64-2.7/numpy/core/src/private -Ibuild/src.linux-x86_64-2.7/numpy/core/src/npymath -c build/src.linux-x86_64-2.7/numpy/core/src/npymath/ieee754.c -o build/temp.linux-x86_64-2.7/build/src.linux-x86_64-2.7/numpy/core/src/npymath/ieee754.o -MMD -MF build/temp.linux-x86_64-2.7/build/src.linux-x86_64-2.7/numpy/core/src/npymath/ieee754.o.d\" failed with exit status 1\r\n```\r\n\r\nSystem information:\r\n\r\n```\r\n$ uname -a\r\nLinux ae-ml-dev 3.10.0-862.3.2.el7.x86_64 #1 SMP Mon May 21 23:36:36 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux\r\n\r\n$ cat /etc/*release*\r\nCentOS Linux release 7.5.1804 (Core)\r\nDerived from Red Hat Enterprise Linux 7.5 (Source)\r\nNAME=\"CentOS Linux\"\r\nVERSION=\"7 (Core)\"\r\nID=\"centos\"\r\nID_LIKE=\"rhel fedora\"\r\nVERSION_ID=\"7\"\r\nPRETTY_NAME=\"CentOS Linux 7 (Core)\"\r\nANSI_COLOR=\"0;31\"\r\nCPE_NAME=\"cpe:/o:centos:centos:7\"\r\nHOME_URL=\"https://www.centos.org/\"\r\nBUG_REPORT_URL=\"https://bugs.centos.org/\"\r\n\r\nCENTOS_MANTISBT_PROJECT=\"CentOS-7\"\r\nCENTOS_MANTISBT_PROJECT_VERSION=\"7\"\r\nREDHAT_SUPPORT_PRODUCT=\"centos\"\r\nREDHAT_SUPPORT_PRODUCT_VERSION=\"7\"\r\n\r\nCentOS Linux release 7.5.1804 (Core)\r\nCentOS Linux release 7.5.1804 (Core)\r\ncpe:/o:centos:centos:7\r\n\r\n$ docker --version\r\nDocker version 18.03.1-ce, build 9ee9f40\r\n```\r\n\r\nFull logs:\r\n\r\n```\r\n$ cd seldon-core/examples/models/sklearn_iris_docker\r\n$ make\r\ndocker build \\\r\n  --build-arg IMAGE_REPO=seldonio \\\r\n  --build-arg IMAGE_VERSION=0.1 \\\r\n  --build-arg MODEL_NAME=IrisClassifier \\\r\n  --tag seldonio/irisclassifier:0.1 .\r\nSending build context to Docker daemon  9.216kB\r\nStep 1/28 : FROM alpine:3.7 as build-alpine\r\n ---> 3fd9065eaf02\r\nStep 2/28 : RUN apk add --update     build-base     freetype-dev     gcc     gfortran     libc6-compat     libffi-dev     libpng-dev     openblas-dev     openssl-dev     py2-pip     python2     python2-dev    wget     && true\r\n ---> Running in 00a7e59023c0\r\nfetch http://dl-cdn.alpinelinux.org/alpine/v3.7/main/x86_64/APKINDEX.tar.gz\r\nfetch http://dl-cdn.alpinelinux.org/alpine/v3.7/community/x86_64/APKINDEX.tar.gz\r\n(1/50) Upgrading musl (1.1.18-r2 -> 1.1.18-r3)\r\n(2/50) Installing binutils-libs (2.30-r1)\r\n(3/50) Installing binutils (2.30-r1)\r\n(4/50) Installing gmp (6.1.2-r1)\r\n(5/50) Installing isl (0.18-r0)\r\n(6/50) Installing libgomp (6.4.0-r5)\r\n(7/50) Installing libatomic (6.4.0-r5)\r\n(8/50) Installing pkgconf (1.3.10-r0)\r\n(9/50) Installing libgcc (6.4.0-r5)\r\n(10/50) Installing mpfr3 (3.1.5-r1)\r\n(11/50) Installing mpc1 (1.0.3-r1)\r\n(12/50) Installing libstdc++ (6.4.0-r5)\r\n(13/50) Installing gcc (6.4.0-r5)\r\n(14/50) Installing musl-dev (1.1.18-r3)\r\n(15/50) Installing libc-dev (0.7.1-r0)\r\n(16/50) Installing g++ (6.4.0-r5)\r\n(17/50) Installing make (4.2.1-r0)\r\n(18/50) Installing fortify-headers (0.9-r0)\r\n(19/50) Installing build-base (0.5-r0)\r\n(20/50) Installing libbz2 (1.0.6-r6)\r\n(21/50) Installing libpng (1.6.34-r1)\r\n(22/50) Installing freetype (2.8.1-r2)\r\n(23/50) Installing zlib-dev (1.2.11-r1)\r\n(24/50) Installing libpng-dev (1.6.34-r1)\r\n(25/50) Installing freetype-dev (2.8.1-r2)\r\n(26/50) Installing libquadmath (6.4.0-r5)\r\n(27/50) Installing libgfortran (6.4.0-r5)\r\n(28/50) Installing gfortran (6.4.0-r5)\r\n(29/50) Upgrading musl-utils (1.1.18-r2 -> 1.1.18-r3)\r\n(30/50) Installing libc6-compat (1.1.18-r3)\r\n(31/50) Installing libffi (3.2.1-r4)\r\n(32/50) Installing libffi-dev (3.2.1-r4)\r\n(33/50) Installing openblas-ilp64 (0.2.19-r3)\r\n(34/50) Installing openblas (0.2.19-r3)\r\n(35/50) Installing openblas-dev (0.2.19-r3)\r\n(36/50) Installing libcrypto1.0 (1.0.2o-r0)\r\n(37/50) Installing libssl1.0 (1.0.2o-r0)\r\n(38/50) Installing openssl-dev (1.0.2o-r0)\r\n(39/50) Installing expat (2.2.5-r0)\r\n(40/50) Installing gdbm (1.13-r1)\r\n(41/50) Installing ncurses-terminfo-base (6.0_p20171125-r0)\r\n(42/50) Installing ncurses-terminfo (6.0_p20171125-r0)\r\n(43/50) Installing ncurses-libs (6.0_p20171125-r0)\r\n(44/50) Installing readline (7.0.003-r0)\r\n(45/50) Installing sqlite-libs (3.21.0-r1)\r\n(46/50) Installing python2 (2.7.14-r2)\r\n(47/50) Installing py-setuptools (33.1.1-r1)\r\n(48/50) Installing py2-pip (9.0.1-r1)\r\n(49/50) Installing python2-dev (2.7.14-r2)\r\n(50/50) Installing wget (1.19.5-r0)\r\nExecuting busybox-1.27.2-r7.trigger\r\nOK: 458 MiB in 59 packages\r\nRemoving intermediate container 00a7e59023c0\r\n ---> 82dbbbf825fc\r\nStep 3/28 : RUN ln -s /usr/include/locale.h /usr/include/xlocale.h\r\n ---> Running in 0f8306aa78eb\r\nRemoving intermediate container 0f8306aa78eb\r\n ---> 064da9b685cf\r\nStep 4/28 : RUN mkdir -p /out/etc/apk && cp -r /etc/apk/* /out/etc/apk/\r\n ---> Running in c3286f8b1ed4\r\nRemoving intermediate container c3286f8b1ed4\r\n ---> a147d72d4f49\r\nStep 5/28 : RUN apk add --no-cache --initdb --root /out     alpine-baselayout     busybox     ca-certificates     freetype     libc6-compat     libffi     libpng     libstdc++     musl     openblas     openssl     python2     && true\r\n ---> Running in afa8bab6266f\r\nfetch http://dl-cdn.alpinelinux.org/alpine/v3.7/main/x86_64/APKINDEX.tar.gz\r\nfetch http://dl-cdn.alpinelinux.org/alpine/v3.7/community/x86_64/APKINDEX.tar.gz\r\n(1/30) Installing musl (1.1.18-r3)\r\n(2/30) Installing busybox (1.27.2-r11)\r\nExecuting busybox-1.27.2-r11.post-install\r\n(3/30) Installing alpine-baselayout (3.0.5-r2)\r\nExecuting alpine-baselayout-3.0.5-r2.pre-install\r\nExecuting alpine-baselayout-3.0.5-r2.post-install\r\n(4/30) Installing libressl2.6-libcrypto (2.6.4-r2)\r\n(5/30) Installing ca-certificates (20171114-r0)\r\n(6/30) Installing libbz2 (1.0.6-r6)\r\n(7/30) Installing zlib (1.2.11-r1)\r\n(8/30) Installing libpng (1.6.34-r1)\r\n(9/30) Installing freetype (2.8.1-r2)\r\n(10/30) Installing libc6-compat (1.1.18-r3)\r\n(11/30) Installing libffi (3.2.1-r4)\r\n(12/30) Installing libgcc (6.4.0-r5)\r\n(13/30) Installing libstdc++ (6.4.0-r5)\r\n(14/30) Installing libquadmath (6.4.0-r5)\r\n(15/30) Installing libgfortran (6.4.0-r5)\r\n(16/30) Installing openblas (0.2.19-r3)\r\n(17/30) Installing libcrypto1.0 (1.0.2o-r0)\r\n(18/30) Installing libssl1.0 (1.0.2o-r0)\r\n(19/30) Installing openssl (1.0.2o-r0)\r\n(20/30) Installing expat (2.2.5-r0)\r\n(21/30) Installing gdbm (1.13-r1)\r\n(22/30) Installing ncurses-terminfo-base (6.0_p20171125-r0)\r\n(23/30) Installing ncurses-terminfo (6.0_p20171125-r0)\r\n(24/30) Installing ncurses-libs (6.0_p20171125-r0)\r\n(25/30) Installing readline (7.0.003-r0)\r\n(26/30) Installing sqlite-libs (3.21.0-r1)\r\n(27/30) Installing libressl2.6-libssl (2.6.4-r2)\r\n(28/30) Installing libressl2.6-libtls (2.6.4-r2)\r\n(29/30) Installing ssl_client (1.27.2-r11)\r\n(30/30) Installing python2 (2.7.14-r2)\r\nExecuting busybox-1.27.2-r11.trigger\r\nExecuting ca-certificates-20171114-r0.trigger\r\nOK: 93 MiB in 30 packages\r\nRemoving intermediate container afa8bab6266f\r\n ---> 986043d44641\r\nStep 6/28 : RUN rm -rf /out/etc/apk /out/lib/apk /out/var/cache\r\n ---> Running in 12c5619c34a8\r\nRemoving intermediate container 12c5619c34a8\r\n ---> 8220d52ace44\r\nStep 7/28 : COPY . /src\r\n ---> 275d11990a10\r\nStep 8/28 : WORKDIR /src\r\nRemoving intermediate container 3c255a6146e6\r\n ---> cf98b7147d55\r\nStep 9/28 : RUN pip install --requirement requirements.txt\r\n ---> Running in fcaf0403e89b\r\nCollecting numpy==1.11.2 (from -r requirements.txt (line 1))\r\n  Downloading https://files.pythonhosted.org/packages/16/f5/b432f028134dd30cfbf6f21b8264a9938e5e0f75204e72453af08d67eb0b/numpy-1.11.2.tar.gz (4.2MB)\r\nCollecting pandas==0.18.1 (from -r requirements.txt (line 2))\r\n  Downloading https://files.pythonhosted.org/packages/11/09/e66eb844daba8680ddff26335d5b4fead77f60f957678243549a8dd4830d/pandas-0.18.1.tar.gz (7.3MB)\r\n    Complete output from command python setup.py egg_info:\r\n    Running from numpy source directory.\r\n    /tmp/easy_install-xjA3le/numpy-1.14.4/setup.py:369: UserWarning: Unrecognized setuptools command, proceeding with generating Cython sources and expanding templates\r\n      run_build = parse_setuppy_commands()\r\n    /bin/sh: svnversion: not found\r\n    non-existing path in 'numpy/distutils': 'site.cfg'\r\n    /bin/sh: svnversion: not found\r\n    /usr/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'define_macros'\r\n      warnings.warn(msg)\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'exp'\r\n     int exp (void);\r\n         ^~~\r\n    _configtest.c:1:24: fatal error: sys/endian.h: No such file or directory\r\n     #include <sys/endian.h>\r\n                            ^\r\n    compilation terminated.\r\n    _configtest.c:1:24: fatal error: sys/endian.h: No such file or directory\r\n     #include <sys/endian.h>\r\n                            ^\r\n    compilation terminated.\r\n    _configtest.c: In function 'main':\r\n    _configtest.c:7:12: error: 'SIZEOF_LONGDOUBLE' undeclared (first use in this function)\r\n         (void) SIZEOF_LONGDOUBLE;\r\n                ^~~~~~~~~~~~~~~~~\r\n    _configtest.c:7:12: note: each undeclared identifier is reported only once for each function it appears in\r\n    _configtest.c: In function 'main':\r\n    _configtest.c:7:12: error: 'SIZEOF_LONGDOUBLE' undeclared (first use in this function)\r\n         (void) SIZEOF_LONGDOUBLE;\r\n                ^~~~~~~~~~~~~~~~~\r\n    _configtest.c:7:12: note: each undeclared identifier is reported only once for each function it appears in\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'exp'\r\n     int exp (void);\r\n         ^~~\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'asin'\r\n     int asin (void);\r\n         ^~~~\r\n    _configtest.c:2:5: warning: conflicting types for built-in function 'cos'\r\n     int cos (void);\r\n         ^~~\r\n    _configtest.c:3:5: warning: conflicting types for built-in function 'log'\r\n     int log (void);\r\n         ^~~\r\n    _configtest.c:4:5: warning: conflicting types for built-in function 'fabs'\r\n     int fabs (void);\r\n         ^~~~\r\n    _configtest.c:5:5: warning: conflicting types for built-in function 'tanh'\r\n     int tanh (void);\r\n         ^~~~\r\n    _configtest.c:6:5: warning: conflicting types for built-in function 'atan'\r\n     int atan (void);\r\n         ^~~~\r\n    _configtest.c:7:5: warning: conflicting types for built-in function 'acos'\r\n     int acos (void);\r\n         ^~~~\r\n    _configtest.c:8:5: warning: conflicting types for built-in function 'floor'\r\n     int floor (void);\r\n         ^~~~~\r\n    _configtest.c:9:5: warning: conflicting types for built-in function 'fmod'\r\n     int fmod (void);\r\n         ^~~~\r\n    _configtest.c:10:5: warning: conflicting types for built-in function 'sqrt'\r\n     int sqrt (void);\r\n         ^~~~\r\n    _configtest.c:11:5: warning: conflicting types for built-in function 'cosh'\r\n     int cosh (void);\r\n         ^~~~\r\n    _configtest.c:12:5: warning: conflicting types for built-in function 'modf'\r\n     int modf (void);\r\n         ^~~~\r\n    _configtest.c:13:5: warning: conflicting types for built-in function 'sinh'\r\n     int sinh (void);\r\n         ^~~~\r\n    _configtest.c:14:5: warning: conflicting types for built-in function 'frexp'\r\n     int frexp (void);\r\n         ^~~~~\r\n    _configtest.c:15:5: warning: conflicting types for built-in function 'exp'\r\n     int exp (void);\r\n         ^~~\r\n    _configtest.c:16:5: warning: conflicting types for built-in function 'tan'\r\n     int tan (void);\r\n         ^~~\r\n    _configtest.c:17:5: warning: conflicting types for built-in function 'ceil'\r\n     int ceil (void);\r\n         ^~~~\r\n    _configtest.c:18:5: warning: conflicting types for built-in function 'log10'\r\n     int log10 (void);\r\n         ^~~~~\r\n    _configtest.c:19:5: warning: conflicting types for built-in function 'sin'\r\n     int sin (void);\r\n         ^~~\r\n    _configtest.c:20:5: warning: conflicting types for built-in function 'ldexp'\r\n     int ldexp (void);\r\n         ^~~~~\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'log2'\r\n     int log2 (void);\r\n         ^~~~\r\n    _configtest.c:2:5: warning: conflicting types for built-in function 'pow'\r\n     int pow (void);\r\n         ^~~\r\n    _configtest.c:4:5: warning: conflicting types for built-in function 'exp2'\r\n     int exp2 (void);\r\n         ^~~~\r\n    _configtest.c:5:5: warning: conflicting types for built-in function 'atan2'\r\n     int atan2 (void);\r\n         ^~~~~\r\n    _configtest.c:7:5: warning: conflicting types for built-in function 'rint'\r\n     int rint (void);\r\n         ^~~~\r\n    _configtest.c:11:5: warning: conflicting types for built-in function 'cbrt'\r\n     int cbrt (void);\r\n         ^~~~\r\n    _configtest.c:12:5: warning: conflicting types for built-in function 'nextafter'\r\n     int nextafter (void);\r\n         ^~~~~~~~~\r\n    _configtest.c:13:5: warning: conflicting types for built-in function 'trunc'\r\n     int trunc (void);\r\n         ^~~~~\r\n    _configtest.o: In function `main':\r\n    /tmp/easy_install-xjA3le/numpy-1.14.4/_configtest.c:42: undefined reference to `backtrace'\r\n    collect2: error: ld returned 1 exit status\r\n    _configtest.o: In function `main':\r\n    /tmp/easy_install-xjA3le/numpy-1.14.4/_configtest.c:42: undefined reference to `backtrace'\r\n    collect2: error: ld returned 1 exit status\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'rint'\r\n     int rint (void);\r\n         ^~~~\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'trunc'\r\n     int trunc (void);\r\n         ^~~~~\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'exp2'\r\n     int exp2 (void);\r\n         ^~~~\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'log2'\r\n     int log2 (void);\r\n         ^~~~\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'atan2'\r\n     int atan2 (void);\r\n         ^~~~~\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'pow'\r\n     int pow (void);\r\n         ^~~\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'nextafter'\r\n     int nextafter (void);\r\n         ^~~~~~~~~\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'cbrt'\r\n     int cbrt (void);\r\n         ^~~~\r\n    _configtest.o: In function `main':\r\n    /tmp/easy_install-xjA3le/numpy-1.14.4/_configtest.c:6: undefined reference to `backtrace'\r\n    collect2: error: ld returned 1 exit status\r\n    _configtest.o: In function `main':\r\n    /tmp/easy_install-xjA3le/numpy-1.14.4/_configtest.c:6: undefined reference to `backtrace'\r\n    collect2: error: ld returned 1 exit status\r\n    _configtest.c: In function '__declspec':\r\n    _configtest.c:5:24: error: expected declaration specifiers before 'foo'\r\n     int __declspec(thread) foo;\r\n                            ^~~\r\n    _configtest.c:9:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token\r\n     {\r\n     ^\r\n    _configtest.c:5:5: warning: type of 'thread' defaults to 'int' [-Wimplicit-int]\r\n     int __declspec(thread) foo;\r\n         ^~~~~~~~~~\r\n    _configtest.c:11:1: error: expected '{' at end of input\r\n     }\r\n     ^\r\n    _configtest.c: In function '__declspec':\r\n    _configtest.c:5:24: error: expected declaration specifiers before 'foo'\r\n     int __declspec(thread) foo;\r\n                            ^~~\r\n    _configtest.c:9:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token\r\n     {\r\n     ^\r\n    _configtest.c:5:5: warning: type of 'thread' defaults to 'int' [-Wimplicit-int]\r\n     int __declspec(thread) foo;\r\n         ^~~~~~~~~~\r\n    _configtest.c:11:1: error: expected '{' at end of input\r\n     }\r\n     ^\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'cosf'\r\n     int cosf (void);\r\n         ^~~~\r\n    _configtest.c:2:5: warning: conflicting types for built-in function 'coshf'\r\n     int coshf (void);\r\n         ^~~~~\r\n    _configtest.c:3:5: warning: conflicting types for built-in function 'rintf'\r\n     int rintf (void);\r\n         ^~~~~\r\n    _configtest.c:4:5: warning: conflicting types for built-in function 'fabsf'\r\n     int fabsf (void);\r\n         ^~~~~\r\n    _configtest.c:5:5: warning: conflicting types for built-in function 'floorf'\r\n     int floorf (void);\r\n         ^~~~~~\r\n    _configtest.c:6:5: warning: conflicting types for built-in function 'nextafterf'\r\n     int nextafterf (void);\r\n         ^~~~~~~~~~\r\n    _configtest.c:7:5: warning: conflicting types for built-in function 'tanhf'\r\n     int tanhf (void);\r\n         ^~~~~\r\n    _configtest.c:8:5: warning: conflicting types for built-in function 'log10f'\r\n     int log10f (void);\r\n         ^~~~~~\r\n    _configtest.c:9:5: warning: conflicting types for built-in function 'logf'\r\n     int logf (void);\r\n         ^~~~\r\n    _configtest.c:10:5: warning: conflicting types for built-in function 'sinhf'\r\n     int sinhf (void);\r\n         ^~~~~\r\n    _configtest.c:11:5: warning: conflicting types for built-in function 'acosf'\r\n     int acosf (void);\r\n         ^~~~~\r\n    _configtest.c:12:5: warning: conflicting types for built-in function 'sqrtf'\r\n     int sqrtf (void);\r\n         ^~~~~\r\n    _configtest.c:13:5: warning: conflicting types for built-in function 'ldexpf'\r\n     int ldexpf (void);\r\n         ^~~~~~\r\n    _configtest.c:14:5: warning: conflicting types for built-in function 'hypotf'\r\n     int hypotf (void);\r\n         ^~~~~~\r\n    _configtest.c:15:5: warning: conflicting types for built-in function 'log2f'\r\n     int log2f (void);\r\n         ^~~~~\r\n    _configtest.c:16:5: warning: conflicting types for built-in function 'exp2f'\r\n     int exp2f (void);\r\n         ^~~~~\r\n    _configtest.c:17:5: warning: conflicting types for built-in function 'atanf'\r\n     int atanf (void);\r\n         ^~~~~\r\n    _configtest.c:18:5: warning: conflicting types for built-in function 'fmodf'\r\n     int fmodf (void);\r\n         ^~~~~\r\n    _configtest.c:19:5: warning: conflicting types for built-in function 'atan2f'\r\n     int atan2f (void);\r\n         ^~~~~~\r\n    _configtest.c:20:5: warning: conflicting types for built-in function 'cbrtf'\r\n     int cbrtf (void);\r\n         ^~~~~\r\n    _configtest.c:21:5: warning: conflicting types for built-in function 'modff'\r\n     int modff (void);\r\n         ^~~~~\r\n    _configtest.c:22:5: warning: conflicting types for built-in function 'ceilf'\r\n     int ceilf (void);\r\n         ^~~~~\r\n    _configtest.c:23:5: warning: conflicting types for built-in function 'log1pf'\r\n     int log1pf (void);\r\n         ^~~~~~\r\n    _configtest.c:24:5: warning: conflicting types for built-in function 'asinf'\r\n     int asinf (void);\r\n         ^~~~~\r\n    _configtest.c:25:5: warning: conflicting types for built-in function 'copysignf'\r\n     int copysignf (void);\r\n         ^~~~~~~~~\r\n    _configtest.c:26:5: warning: conflicting types for built-in function 'acoshf'\r\n     int acoshf (void);\r\n         ^~~~~~\r\n    _configtest.c:27:5: warning: conflicting types for built-in function 'sinf'\r\n     int sinf (void);\r\n         ^~~~\r\n    _configtest.c:28:5: warning: conflicting types for built-in function 'tanf'\r\n     int tanf (void);\r\n         ^~~~\r\n    _configtest.c:29:5: warning: conflicting types for built-in function 'atanhf'\r\n     int atanhf (void);\r\n         ^~~~~~\r\n    _configtest.c:30:5: warning: conflicting types for built-in function 'truncf'\r\n     int truncf (void);\r\n         ^~~~~~\r\n    _configtest.c:31:5: warning: conflicting types for built-in function 'asinhf'\r\n     int asinhf (void);\r\n         ^~~~~~\r\n    _configtest.c:32:5: warning: conflicting types for built-in function 'frexpf'\r\n     int frexpf (void);\r\n         ^~~~~~\r\n    _configtest.c:33:5: warning: conflicting types for built-in function 'powf'\r\n     int powf (void);\r\n         ^~~~\r\n    _configtest.c:34:5: warning: conflicting types for built-in function 'expf'\r\n     int expf (void);\r\n         ^~~~\r\n    _configtest.c:35:5: warning: conflicting types for built-in function 'expm1f'\r\n     int expm1f (void);\r\n         ^~~~~~\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'tanhl'\r\n     int tanhl (void);\r\n         ^~~~~\r\n    _configtest.c:2:5: warning: conflicting types for built-in function 'log10l'\r\n     int log10l (void);\r\n         ^~~~~~\r\n    _configtest.c:3:5: warning: conflicting types for built-in function 'nextafterl'\r\n     int nextafterl (void);\r\n         ^~~~~~~~~~\r\n    _configtest.c:4:5: warning: conflicting types for built-in function 'coshl'\r\n     int coshl (void);\r\n         ^~~~~\r\n    _configtest.c:5:5: warning: conflicting types for built-in function 'cosl'\r\n     int cosl (void);\r\n         ^~~~\r\n    _configtest.c:6:5: warning: conflicting types for built-in function 'floorl'\r\n     int floorl (void);\r\n         ^~~~~~\r\n    _configtest.c:7:5: warning: conflicting types for built-in function 'rintl'\r\n     int rintl (void);\r\n         ^~~~~\r\n    _configtest.c:8:5: warning: conflicting types for built-in function 'fabsl'\r\n     int fabsl (void);\r\n         ^~~~~\r\n    _configtest.c:9:5: warning: conflicting types for built-in function 'acosl'\r\n     int acosl (void);\r\n         ^~~~~\r\n    _configtest.c:10:5: warning: conflicting types for built-in function 'ldexpl'\r\n     int ldexpl (void);\r\n         ^~~~~~\r\n    _configtest.c:11:5: warning: conflicting types for built-in function 'sqrtl'\r\n     int sqrtl (void);\r\n         ^~~~~\r\n    _configtest.c:12:5: warning: conflicting types for built-in function 'logl'\r\n     int logl (void);\r\n         ^~~~\r\n    _configtest.c:13:5: warning: conflicting types for built-in function 'expm1l'\r\n     int expm1l (void);\r\n         ^~~~~~\r\n    _configtest.c:14:5: warning: conflicting types for built-in function 'hypotl'\r\n     int hypotl (void);\r\n         ^~~~~~\r\n    _configtest.c:15:5: warning: conflicting types for built-in function 'log2l'\r\n     int log2l (void);\r\n         ^~~~~\r\n    _configtest.c:16:5: warning: conflicting types for built-in function 'copysignl'\r\n     int copysignl (void);\r\n         ^~~~~~~~~\r\n    _configtest.c:17:5: warning: conflicting types for built-in function 'exp2l'\r\n     int exp2l (void);\r\n         ^~~~~\r\n    _configtest.c:18:5: warning: conflicting types for built-in function 'atanl'\r\n     int atanl (void);\r\n         ^~~~~\r\n    _configtest.c:19:5: warning: conflicting types for built-in function 'frexpl'\r\n     int frexpl (void);\r\n         ^~~~~~\r\n    _configtest.c:20:5: warning: conflicting types for built-in function 'atan2l'\r\n     int atan2l (void);\r\n         ^~~~~~\r\n    _configtest.c:21:5: warning: conflicting types for built-in function 'sinhl'\r\n     int sinhl (void);\r\n         ^~~~~\r\n    _configtest.c:22:5: warning: conflicting types for built-in function 'fmodl'\r\n     int fmodl (void);\r\n         ^~~~~\r\n    _configtest.c:23:5: warning: conflicting types for built-in function 'log1pl'\r\n     int log1pl (void);\r\n         ^~~~~~\r\n    _configtest.c:24:5: warning: conflicting types for built-in function 'asinl'\r\n     int asinl (void);\r\n         ^~~~~\r\n    _configtest.c:25:5: warning: conflicting types for built-in function 'ceill'\r\n     int ceill (void);\r\n         ^~~~~\r\n    _configtest.c:26:5: warning: conflicting types for built-in function 'sinl'\r\n     int sinl (void);\r\n         ^~~~\r\n    _configtest.c:27:5: warning: conflicting types for built-in function 'cbrtl'\r\n     int cbrtl (void);\r\n         ^~~~~\r\n    _configtest.c:28:5: warning: conflicting types for built-in function 'acoshl'\r\n     int acoshl (void);\r\n         ^~~~~~\r\n    _configtest.c:29:5: warning: conflicting types for built-in function 'atanhl'\r\n     int atanhl (void);\r\n         ^~~~~~\r\n    _configtest.c:30:5: warning: conflicting types for built-in function 'tanl'\r\n     int tanl (void);\r\n         ^~~~\r\n    _configtest.c:31:5: warning: conflicting types for built-in function 'truncl'\r\n     int truncl (void);\r\n         ^~~~~~\r\n    _configtest.c:32:5: warning: conflicting types for built-in function 'powl'\r\n     int powl (void);\r\n         ^~~~\r\n    _configtest.c:33:5: warning: conflicting types for built-in function 'expl'\r\n     int expl (void);\r\n         ^~~~\r\n    _configtest.c:34:5: warning: conflicting types for built-in function 'modfl'\r\n     int modfl (void);\r\n         ^~~~~\r\n    _configtest.c:35:5: warning: conflicting types for built-in function 'asinhl'\r\n     int asinhl (void);\r\n         ^~~~~~\r\n    _configtest.c: In function 'main':\r\n    _configtest.c:8:12: error: 'HAVE_DECL_SIGNBIT' undeclared (first use in this function)\r\n         (void) HAVE_DECL_SIGNBIT;\r\n                ^~~~~~~~~~~~~~~~~\r\n    _configtest.c:8:12: note: each undeclared identifier is reported only once for each function it appears in\r\n    _configtest.c: In function 'main':\r\n    _configtest.c:8:12: error: 'HAVE_DECL_SIGNBIT' undeclared (first use in this function)\r\n         (void) HAVE_DECL_SIGNBIT;\r\n                ^~~~~~~~~~~~~~~~~\r\n    _configtest.c:8:12: note: each undeclared identifier is reported only once for each function it appears in\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'clog'\r\n     int clog (void);\r\n         ^~~~\r\n    _configtest.c:2:5: warning: conflicting types for built-in function 'ctan'\r\n     int ctan (void);\r\n         ^~~~\r\n    _configtest.c:3:5: warning: conflicting types for built-in function 'casinh'\r\n     int casinh (void);\r\n         ^~~~~~\r\n    _configtest.c:4:5: warning: conflicting types for built-in function 'cacosh'\r\n     int cacosh (void);\r\n         ^~~~~~\r\n    _configtest.c:5:5: warning: conflicting types for built-in function 'cabs'\r\n     int cabs (void);\r\n         ^~~~\r\n    _configtest.c:6:5: warning: conflicting types for built-in function 'conj'\r\n     int conj (void);\r\n         ^~~~\r\n    _configtest.c:7:5: warning: conflicting types for built-in function 'csqrt'\r\n     int csqrt (void);\r\n         ^~~~~\r\n    _configtest.c:8:5: warning: conflicting types for built-in function 'creal'\r\n     int creal (void);\r\n         ^~~~~\r\n    _configtest.c:9:5: warning: conflicting types for built-in function 'cproj'\r\n     int cproj (void);\r\n         ^~~~~\r\n    _configtest.c:10:5: warning: conflicting types for built-in function 'cpow'\r\n     int cpow (void);\r\n         ^~~~\r\n    _configtest.c:11:5: warning: conflicting types for built-in function 'csinh'\r\n     int csinh (void);\r\n         ^~~~~\r\n    _configtest.c:12:5: warning: conflicting types for built-in function 'catan'\r\n     int catan (void);\r\n         ^~~~~\r\n    _configtest.c:13:5: warning: conflicting types for built-in function 'ctanh'\r\n     int ctanh (void);\r\n         ^~~~~\r\n    _configtest.c:14:5: warning: conflicting types for built-in function 'ccosh'\r\n     int ccosh (void);\r\n         ^~~~~\r\n    _configtest.c:15:5: warning: conflicting types for built-in function 'cimag'\r\n     int cimag (void);\r\n         ^~~~~\r\n    _configtest.c:16:5: warning: conflicting types for built-in function 'casin'\r\n     int casin (void);\r\n         ^~~~~\r\n    _configtest.c:17:5: warning: conflicting types for built-in function 'carg'\r\n     int carg (void);\r\n         ^~~~\r\n    _configtest.c:18:5: warning: conflicting types for built-in function 'csin'\r\n     int csin (void);\r\n         ^~~~\r\n    _configtest.c:19:5: warning: conflicting types for built-in function 'cexp'\r\n     int cexp (void);\r\n         ^~~~\r\n    _configtest.c:20:5: warning: conflicting types for built-in function 'ccos'\r\n     int ccos (void);\r\n         ^~~~\r\n    _configtest.c:21:5: warning: conflicting types for built-in function 'cacos'\r\n     int cacos (void);\r\n         ^~~~~\r\n    _configtest.c:22:5: warning: conflicting types for built-in function 'catanh'\r\n     int catanh (void);\r\n         ^~~~~~\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'ctanhf'\r\n     int ctanhf (void);\r\n         ^~~~~~\r\n    _configtest.c:2:5: warning: conflicting types for built-in function 'csinhf'\r\n     int csinhf (void);\r\n         ^~~~~~\r\n    _configtest.c:3:5: warning: conflicting types for built-in function 'cprojf'\r\n     int cprojf (void);\r\n         ^~~~~~\r\n    _configtest.c:4:5: warning: conflicting types for built-in function 'cacoshf'\r\n     int cacoshf (void);\r\n         ^~~~~~~\r\n    _configtest.c:5:5: warning: conflicting types for built-in function 'cacosf'\r\n     int cacosf (void);\r\n         ^~~~~~\r\n    _configtest.c:6:5: warning: conflicting types for built-in function 'cargf'\r\n     int cargf (void);\r\n         ^~~~~\r\n    _configtest.c:7:5: warning: conflicting types for built-in function 'cpowf'\r\n     int cpowf (void);\r\n         ^~~~~\r\n    _configtest.c:8:5: warning: conflicting types for built-in function 'csinf'\r\n     int csinf (void);\r\n         ^~~~~\r\n    _configtest.c:9:5: warning: conflicting types for built-in function 'clogf'\r\n     int clogf (void);\r\n         ^~~~~\r\n    _configtest.c:10:5: warning: conflicting types for built-in function 'ccoshf'\r\n     int ccoshf (void);\r\n         ^~~~~~\r\n    _configtest.c:11:5: warning: conflicting types for built-in function 'conjf'\r\n     int conjf (void);\r\n         ^~~~~\r\n    _configtest.c:12:5: warning: conflicting types for built-in function 'casinhf'\r\n     int casinhf (void);\r\n         ^~~~~~~\r\n    _configtest.c:13:5: warning: conflicting types for built-in function 'ccosf'\r\n     int ccosf (void);\r\n         ^~~~~\r\n    _configtest.c:14:5: warning: conflicting types for built-in function 'ctanf'\r\n     int ctanf (void);\r\n         ^~~~~\r\n    _configtest.c:15:5: warning: conflicting types for built-in function 'cexpf'\r\n     int cexpf (void);\r\n         ^~~~~\r\n    _configtest.c:16:5: warning: conflicting types for built-in function 'cimagf'\r\n     int cimagf (void);\r\n         ^~~~~~\r\n    _configtest.c:17:5: warning: conflicting types for built-in function 'catanhf'\r\n     int catanhf (void);\r\n         ^~~~~~~\r\n    _configtest.c:18:5: warning: conflicting types for built-in function 'casinf'\r\n     int casinf (void);\r\n         ^~~~~~\r\n    _configtest.c:19:5: warning: conflicting types for built-in function 'catanf'\r\n     int catanf (void);\r\n         ^~~~~~\r\n    _configtest.c:20:5: warning: conflicting types for built-in function 'csqrtf'\r\n     int csqrtf (void);\r\n         ^~~~~~\r\n    _configtest.c:21:5: warning: conflicting types for built-in function 'crealf'\r\n     int crealf (void);\r\n         ^~~~~~\r\n    _configtest.c:22:5: warning: conflicting types for built-in function 'cabsf'\r\n     int cabsf (void);\r\n         ^~~~~\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'csinhl'\r\n     int csinhl (void);\r\n         ^~~~~~\r\n    _configtest.c:2:5: warning: conflicting types for built-in function 'ctanhl'\r\n     int ctanhl (void);\r\n         ^~~~~~\r\n    _configtest.c:3:5: warning: conflicting types for built-in function 'cacosl'\r\n     int cacosl (void);\r\n         ^~~~~~\r\n    _configtest.c:4:5: warning: conflicting types for built-in function 'cacoshl'\r\n     int cacoshl (void);\r\n         ^~~~~~~\r\n    _configtest.c:5:5: warning: conflicting types for built-in function 'cprojl'\r\n     int cprojl (void);\r\n         ^~~~~~\r\n    _configtest.c:6:5: warning: conflicting types for built-in function 'cargl'\r\n     int cargl (void);\r\n         ^~~~~\r\n    _configtest.c:7:5: warning: conflicting types for built-in function 'ccoshl'\r\n     int ccoshl (void);\r\n         ^~~~~~\r\n    _configtest.c:8:5: warning: conflicting types for built-in function 'conjl'\r\n     int conjl (void);\r\n         ^~~~~\r\n    _configtest.c:9:5: warning: conflicting types for built-in function 'csinl'\r\n     int csinl (void);\r\n         ^~~~~\r\n    _configtest.c:10:5: warning: conflicting types for built-in function 'clogl'\r\n     int clogl (void);\r\n         ^~~~~\r\n    _configtest.c:11:5: warning: conflicting types for built-in function 'cexpl'\r\n     int cexpl (void);\r\n         ^~~~~\r\n    _configtest.c:12:5: warning: conflicting types for built-in function 'ccosl'\r\n     int ccosl (void);\r\n         ^~~~~\r\n    _configtest.c:13:5: warning: conflicting types for built-in function 'ctanl'\r\n     int ctanl (void);\r\n         ^~~~~\r\n    _configtest.c:14:5: warning: conflicting types for built-in function 'cpowl'\r\n     int cpowl (void);\r\n         ^~~~~\r\n    _configtest.c:15:5: warning: conflicting types for built-in function 'casinhl'\r\n     int casinhl (void);\r\n         ^~~~~~~\r\n    _configtest.c:16:5: warning: conflicting types for built-in function 'csqrtl'\r\n     int csqrtl (void);\r\n         ^~~~~~\r\n    _configtest.c:17:5: warning: conflicting types for built-in function 'casinl'\r\n     int casinl (void);\r\n         ^~~~~~\r\n    _configtest.c:18:5: warning: conflicting types for built-in function 'catanl'\r\n     int catanl (void);\r\n         ^~~~~~\r\n    _configtest.c:19:5: warning: conflicting types for built-in function 'catanhl'\r\n     int catanhl (void);\r\n         ^~~~~~~\r\n    _configtest.c:20:5: warning: conflicting types for built-in function 'creall'\r\n     int creall (void);\r\n         ^~~~~~\r\n    _configtest.c:21:5: warning: conflicting types for built-in function 'cabsl'\r\n     int cabsl (void);\r\n         ^~~~~\r\n    _configtest.c:22:5: warning: conflicting types for built-in function 'cimagl'\r\n     int cimagl (void);\r\n         ^~~~~~\r\n    File: build/src.linux-x86_64-2.7/numpy/core/include/numpy/config.h\r\n    #define HAVE_ENDIAN_H 1\r\n    #define SIZEOF_PY_INTPTR_T 8\r\n    #define SIZEOF_OFF_T 8\r\n    #define SIZEOF_PY_LONG_LONG 8\r\n    #define MATHLIB\r\n    #define HAVE_SIN 1\r\n    #define HAVE_COS 1\r\n    #define HAVE_TAN 1\r\n    #define HAVE_SINH 1\r\n    #define HAVE_COSH 1\r\n    #define HAVE_TANH 1\r\n    #define HAVE_FABS 1\r\n    #define HAVE_FLOOR 1\r\n    #define HAVE_CEIL 1\r\n    #define HAVE_SQRT 1\r\n    #define HAVE_LOG10 1\r\n    #define HAVE_LOG 1\r\n    #define HAVE_EXP 1\r\n    #define HAVE_ASIN 1\r\n    #define HAVE_ACOS 1\r\n    #define HAVE_ATAN 1\r\n    #define HAVE_FMOD 1\r\n    #define HAVE_MODF 1\r\n    #define HAVE_FREXP 1\r\n    #define HAVE_LDEXP 1\r\n    #define HAVE_RINT 1\r\n    #define HAVE_TRUNC 1\r\n    #define HAVE_EXP2 1\r\n    #define HAVE_LOG2 1\r\n    #define HAVE_ATAN2 1\r\n    #define HAVE_POW 1\r\n    #define HAVE_NEXTAFTER 1\r\n    #define HAVE_STRTOLL 1\r\n    #define HAVE_STRTOULL 1\r\n    #define HAVE_CBRT 1\r\n    #define HAVE_STRTOLD_L 1\r\n    #define HAVE_FALLOCATE 1\r\n    #define HAVE_XMMINTRIN_H 1\r\n    #define HAVE_EMMINTRIN_H 1\r\n    #define HAVE_FEATURES_H 1\r\n    #define HAVE_XLOCALE_H 1\r\n    #define HAVE_DLFCN_H 1\r\n    #define HAVE___BUILTIN_ISNAN 1\r\n    #define HAVE___BUILTIN_ISINF 1\r\n    #define HAVE___BUILTIN_ISFINITE 1\r\n    #define HAVE___BUILTIN_BSWAP32 1\r\n    #define HAVE___BUILTIN_BSWAP64 1\r\n    #define HAVE___BUILTIN_EXPECT 1\r\n    #define HAVE___BUILTIN_MUL_OVERFLOW 1\r\n    #define HAVE___BUILTIN_CPU_SUPPORTS 1\r\n    #define HAVE__M_FROM_INT64 1\r\n    #define HAVE__MM_LOAD_PS 1\r\n    #define HAVE__MM_PREFETCH 1\r\n    #define HAVE__MM_LOAD_PD 1\r\n    #define HAVE___BUILTIN_PREFETCH 1\r\n    #define HAVE_LINK_AVX 1\r\n    #define HAVE_LINK_AVX2 1\r\n    #define HAVE_ATTRIBUTE_OPTIMIZE_UNROLL_LOOPS 1\r\n    #define HAVE_ATTRIBUTE_OPTIMIZE_OPT_3 1\r\n    #define HAVE_ATTRIBUTE_NONNULL 1\r\n    #define HAVE_ATTRIBUTE_TARGET_AVX 1\r\n    #define HAVE_ATTRIBUTE_TARGET_AVX2 1\r\n    #define HAVE___THREAD 1\r\n    #define HAVE_SINF 1\r\n    #define HAVE_COSF 1\r\n    #define HAVE_TANF 1\r\n    #define HAVE_SINHF 1\r\n    #define HAVE_COSHF 1\r\n    #define HAVE_TANHF 1\r\n    #define HAVE_FABSF 1\r\n    #define HAVE_FLOORF 1\r\n    #define HAVE_CEILF 1\r\n    #define HAVE_RINTF 1\r\n    #define HAVE_TRUNCF 1\r\n    #define HAVE_SQRTF 1\r\n    #define HAVE_LOG10F 1\r\n    #define HAVE_LOGF 1\r\n    #define HAVE_LOG1PF 1\r\n    #define HAVE_EXPF 1\r\n    #define HAVE_EXPM1F 1\r\n    #define HAVE_ASINF 1\r\n    #define HAVE_ACOSF 1\r\n    #define HAVE_ATANF 1\r\n    #define HAVE_ASINHF 1\r\n    #define HAVE_ACOSHF 1\r\n    #define HAVE_ATANHF 1\r\n    #define HAVE_HYPOTF 1\r\n    #define HAVE_ATAN2F 1\r\n    #define HAVE_POWF 1\r\n    #define HAVE_FMODF 1\r\n    #define HAVE_MODFF 1\r\n    #define HAVE_FREXPF 1\r\n    #define HAVE_LDEXPF 1\r\n    #define HAVE_EXP2F 1\r\n    #define HAVE_LOG2F 1\r\n    #define HAVE_COPYSIGNF 1\r\n    #define HAVE_NEXTAFTERF 1\r\n    #define HAVE_CBRTF 1\r\n    #define HAVE_SINL 1\r\n    #define HAVE_COSL 1\r\n    #define HAVE_TANL 1\r\n    #define HAVE_SINHL 1\r\n    #define HAVE_COSHL 1\r\n    #define HAVE_TANHL 1\r\n    #define HAVE_FABSL 1\r\n    #define HAVE_FLOORL 1\r\n    #define HAVE_CEILL 1\r\n    #define HAVE_RINTL 1\r\n    #define HAVE_TRUNCL 1\r\n    #define HAVE_SQRTL 1\r\n    #define HAVE_LOG10L 1\r\n    #define HAVE_LOGL 1\r\n    #define HAVE_LOG1PL 1\r\n    #define HAVE_EXPL 1\r\n    #define HAVE_EXPM1L 1\r\n    #define HAVE_ASINL 1\r\n    #define HAVE_ACOSL 1\r\n    #define HAVE_ATANL 1\r\n    #define HAVE_ASINHL 1\r\n    #define HAVE_ACOSHL 1\r\n    #define HAVE_ATANHL 1\r\n    #define HAVE_HYPOTL 1\r\n    #define HAVE_ATAN2L 1\r\n    #define HAVE_POWL 1\r\n    #define HAVE_FMODL 1\r\n    #define HAVE_MODFL 1\r\n    #define HAVE_FREXPL 1\r\n    #define HAVE_LDEXPL 1\r\n    #define HAVE_EXP2L 1\r\n    #define HAVE_LOG2L 1\r\n    #define HAVE_COPYSIGNL 1\r\n    #define HAVE_NEXTAFTERL 1\r\n    #define HAVE_CBRTL 1\r\n    #define HAVE_DECL_SIGNBIT\r\n    #define HAVE_COMPLEX_H 1\r\n    #define HAVE_CABS 1\r\n    #define HAVE_CACOS 1\r\n    #define HAVE_CACOSH 1\r\n    #define HAVE_CARG 1\r\n    #define HAVE_CASIN 1\r\n    #define HAVE_CASINH 1\r\n    #define HAVE_CATAN 1\r\n    #define HAVE_CATANH 1\r\n    #define HAVE_CCOS 1\r\n    #define HAVE_CCOSH 1\r\n    #define HAVE_CEXP 1\r\n    #define HAVE_CIMAG 1\r\n    #define HAVE_CLOG 1\r\n    #define HAVE_CONJ 1\r\n    #define HAVE_CPOW 1\r\n    #define HAVE_CPROJ 1\r\n    #define HAVE_CREAL 1\r\n    #define HAVE_CSIN 1\r\n    #define HAVE_CSINH 1\r\n    #define HAVE_CSQRT 1\r\n    #define HAVE_CTAN 1\r\n    #define HAVE_CTANH 1\r\n    #define HAVE_CABSF 1\r\n    #define HAVE_CACOSF 1\r\n    #define HAVE_CACOSHF 1\r\n    #define HAVE_CARGF 1\r\n    #define HAVE_CASINF 1\r\n    #define HAVE_CASINHF 1\r\n    #define HAVE_CATANF 1\r\n    #define HAVE_CATANHF 1\r\n    #define HAVE_CCOSF 1\r\n    #define HAVE_CCOSHF 1\r\n    #define HAVE_CEXPF 1\r\n    #define HAVE_CIMAGF 1\r\n    #define HAVE_CLOGF 1\r\n    #define HAVE_CONJF 1\r\n    #define HAVE_CPOWF 1\r\n    #define HAVE_CPROJF 1\r\n    #define HAVE_CREALF 1\r\n    #define HAVE_CSINF 1\r\n    #define HAVE_CSINHF 1\r\n    #define HAVE_CSQRTF 1\r\n    #define HAVE_CTANF 1\r\n    #define HAVE_CTANHF 1\r\n    #define HAVE_CABSL 1\r\n    #define HAVE_CACOSL 1\r\n    #define HAVE_CACOSHL 1\r\n    #define HAVE_CARGL 1\r\n    #define HAVE_CASINL 1\r\n    #define HAVE_CASINHL 1\r\n    #define HAVE_CATANL 1\r\n    #define HAVE_CATANHL 1\r\n    #define HAVE_CCOSL 1\r\n    #define HAVE_CCOSHL 1\r\n    #define HAVE_CEXPL 1\r\n    #define HAVE_CIMAGL 1\r\n    #define HAVE_CLOGL 1\r\n    #define HAVE_CONJL 1\r\n    #define HAVE_CPOWL 1\r\n    #define HAVE_CPROJL 1\r\n    #define HAVE_CREALL 1\r\n    #define HAVE_CSINL 1\r\n    #define HAVE_CSINHL 1\r\n    #define HAVE_CSQRTL 1\r\n    #define HAVE_CTANL 1\r\n    #define HAVE_CTANHL 1\r\n    #define NPY_RESTRICT restrict\r\n    #define NPY_RELAXED_STRIDES_CHECKING 1\r\n    #define HAVE_LDOUBLE_INTEL_EXTENDED_16_BYTES_LE 1\r\n    #ifndef __cplusplus\r\n    /* #undef inline */\r\n    #endif\r\n\r\n    #ifndef _NPY_NPY_CONFIG_H_\r\n    #error config.h should never be included directly, include npy_config.h instead\r\n    #endif\r\n\r\n    EOF\r\n    _configtest.c:1:5: warning: conflicting types for built-in function 'exp'\r\n     int exp (void);\r\n         ^~~\r\n    File: build/src.linux-x86_64-2.7/numpy/core/include/numpy/_numpyconfig.h\r\n    #define NPY_HAVE_ENDIAN_H 1\r\n    #define NPY_SIZEOF_SHORT SIZEOF_SHORT\r\n    #define NPY_SIZEOF_INT SIZEOF_INT\r\n    #define NPY_SIZEOF_LONG SIZEOF_LONG\r\n    #define NPY_SIZEOF_FLOAT 4\r\n    #define NPY_SIZEOF_COMPLEX_FLOAT 8\r\n    #define NPY_SIZEOF_DOUBLE 8\r\n    #define NPY_SIZEOF_COMPLEX_DOUBLE 16\r\n    #define NPY_SIZEOF_LONGDOUBLE 16\r\n    #define NPY_SIZEOF_COMPLEX_LONGDOUBLE 32\r\n    #define NPY_SIZEOF_PY_INTPTR_T 8\r\n    #define NPY_SIZEOF_OFF_T 8\r\n    #define NPY_SIZEOF_PY_LONG_LONG 8\r\n    #define NPY_SIZEOF_LONGLONG 8\r\n    #define NPY_NO_SMP 0\r\n    #define NPY_HAVE_DECL_ISNAN\r\n    #define NPY_HAVE_DECL_ISINF\r\n    #define NPY_HAVE_DECL_ISFINITE\r\n    #define NPY_HAVE_DECL_SIGNBIT\r\n    #define NPY_USE_C99_COMPLEX 1\r\n    #define NPY_HAVE_COMPLEX_DOUBLE 1\r\n    #define NPY_HAVE_COMPLEX_FLOAT 1\r\n    #define NPY_HAVE_COMPLEX_LONG_DOUBLE 1\r\n    #define NPY_RELAXED_STRIDES_CHECKING 1\r\n    #define NPY_USE_C99_FORMATS 1\r\n    #define NPY_VISIBILITY_HIDDEN __attribute__((visibility(\"hidden\")))\r\n    #define NPY_ABI_VERSION 0x01000009\r\n    #define NPY_API_VERSION 0x0000000C\r\n\r\n    #ifndef __STDC_FORMAT_MACROS\r\n    #define __STDC_FORMAT_MACROS 1\r\n    #endif\r\n\r\n    EOF\r\n    warning: no previously-included files matching '*.pyo' found anywhere in distribution\r\n    warning: no previously-included files matching '*.pyd' found anywhere in distribution\r\n    warning: no previously-included files matching '*.swp' found anywhere in distribution\r\n    warning: no previously-included files matching '*.bak' found anywhere in distribution\r\n    warning: no previously-included files matching '*~' found anywhere in distribution\r\n    In file included from numpy/core/src/npymath/ieee754.c.src:9:0:\r\n    numpy/core/include/numpy/utils.h:17:24: error: pasting \"__NPY_UNUSED_TAGGED\" and \"*\" does not give a valid preprocessing token\r\n     #define NPY_UNUSED(x) (__NPY_UNUSED_TAGGED ## x) __COMP_NPY_UNUSED\r\n                            ^\r\n    numpy/core/src/npymath/ieee754.c.src:802:38: note: in expansion of macro 'NPY_UNUSED'\r\n     int npy_get_floatstatus_barrier(char NPY_UNUSED(*param))\r\n                                          ^~~~~~~~~~\r\n    numpy/core/src/npymath/ieee754.c.src:802:49: error: expected ')' before '*' token\r\n     int npy_get_floatstatus_barrier(char NPY_UNUSED(*param))\r\n                                                     ^\r\n    numpy/core/include/numpy/utils.h:17:47: note: in definition of macro 'NPY_UNUSED'\r\n     #define NPY_UNUSED(x) (__NPY_UNUSED_TAGGED ## x) __COMP_NPY_UNUSED\r\n                                                   ^\r\n    In file included from numpy/core/src/npymath/ieee754.c.src:9:0:\r\n    numpy/core/include/numpy/utils.h:17:24: error: pasting \"__NPY_UNUSED_TAGGED\" and \"*\" does not give a valid preprocessing token\r\n     #define NPY_UNUSED(x) (__NPY_UNUSED_TAGGED ## x) __COMP_NPY_UNUSED\r\n                            ^\r\n    numpy/core/src/npymath/ieee754.c.src:802:38: note: in expansion of macro 'NPY_UNUSED'\r\n     int npy_get_floatstatus_barrier(char NPY_UNUSED(*param))\r\n                                          ^~~~~~~~~~\r\n    numpy/core/src/npymath/ieee754.c.src:802:49: error: expected ')' before '*' token\r\n     int npy_get_floatstatus_barrier(char NPY_UNUSED(*param))\r\n                                                     ^\r\n    numpy/core/include/numpy/utils.h:17:47: note: in definition of macro 'NPY_UNUSED'\r\n     #define NPY_UNUSED(x) (__NPY_UNUSED_TAGGED ## x) __COMP_NPY_UNUSED\r\n                                                   ^\r\n    Traceback (most recent call last):\r\n      File \"<string>\", line 1, in <module>\r\n      File \"/tmp/pip-build-fUc8dq/pandas/setup.py\", line 631, in <module>\r\n        **setuptools_kwargs)\r\n      File \"/usr/lib/python2.7/distutils/core.py\", line 111, in setup\r\n        _setup_distribution = dist = klass(attrs)\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/dist.py\", line 317, in __init__\r\n        self.fetch_build_eggs(attrs['setup_requires'])\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/dist.py\", line 372, in fetch_build_eggs\r\n        replace_conflicting=True,\r\n      File \"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py\", line 846, in resolve\r\n        dist = best[req.key] = env.best_match(req, ws, installer)\r\n      File \"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py\", line 1118, in best_match\r\n        return self.obtain(req, installer)\r\n      File \"/usr/lib/python2.7/site-packages/pkg_resources/__init__.py\", line 1130, in obtain\r\n        return installer(requirement)\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/dist.py\", line 440, in fetch_build_egg\r\n        return cmd.easy_install(req)\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py\", line 674, in easy_install\r\n        return self.install_item(spec, dist.location, tmpdir, deps)\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py\", line 700, in install_item\r\n        dists = self.install_eggs(spec, download, tmpdir)\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py\", line 881, in install_eggs\r\n        return self.build_and_install(setup_script, setup_base)\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py\", line 1120, in build_and_install\r\n        self.run_setup(setup_script, setup_base, args)\r\n      File \"/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py\", line 1108, in run_setup\r\n        raise DistutilsError(\"Setup script exited with %s\" % (v.args[0],))\r\n    distutils.errors.DistutilsError: Setup script exited with error: Command \"gcc -fno-strict-aliasing -Os -fomit-frame-pointer -g -DNDEBUG -Os -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x100000 -fPIC -Ibuild/src.linux-x86_64-2.7/numpy/core/src/npymath -Inumpy/core/include -Ibuild/src.linux-x86_64-2.7/numpy/core/include/numpy -Inumpy/core/src/private -Inumpy/core/src -Inumpy/core -Inumpy/core/src/npymath -Inumpy/core/src/multiarray -Inumpy/core/src/umath -Inumpy/core/src/npysort -I/usr/include/python2.7 -Ibuild/src.linux-x86_64-2.7/numpy/core/src/private -Ibuild/src.linux-x86_64-2.7/numpy/core/src/npymath -Ibuild/src.linux-x86_64-2.7/numpy/core/src/private -Ibuild/src.linux-x86_64-2.7/numpy/core/src/npymath -Ibuild/src.linux-x86_64-2.7/numpy/core/src/private -Ibuild/src.linux-x86_64-2.7/numpy/core/src/npymath -c build/src.linux-x86_64-2.7/numpy/core/src/npymath/ieee754.c -o build/temp.linux-x86_64-2.7/build/src.linux-x86_64-2.7/numpy/core/src/npymath/ieee754.o -MMD -MF build/temp.linux-x86_64-2.7/build/src.linux-x86_64-2.7/numpy/core/src/npymath/ieee754.o.d\" failed with exit status 1\r\n\r\n    ----------------------------------------\r\nCommand \"python setup.py egg_info\" failed with error code 1 in /tmp/pip-build-fUc8dq/pandas/\r\nYou are using pip version 9.0.1, however version 10.0.1 is available.\r\nYou should consider upgrading via the 'pip install --upgrade pip' command.\r\nThe command '/bin/sh -c pip install --requirement requirements.txt' returned a non-zero code: 1\r\nmake: *** [container_image] Error 1\r\n```",
      "score": 69.12406
    },
    {
      "url": "https://api.github.com/repos/faizann24/Fwaf-Machine-Learning-driven-Web-Application-Firewall/issues/8",
      "repository_url": "https://api.github.com/repos/faizann24/Fwaf-Machine-Learning-driven-Web-Application-Firewall",
      "labels_url": "https://api.github.com/repos/faizann24/Fwaf-Machine-Learning-driven-Web-Application-Firewall/issues/8/labels{/name}",
      "comments_url": "https://api.github.com/repos/faizann24/Fwaf-Machine-Learning-driven-Web-Application-Firewall/issues/8/comments",
      "events_url": "https://api.github.com/repos/faizann24/Fwaf-Machine-Learning-driven-Web-Application-Firewall/issues/8/events",
      "html_url": "https://github.com/faizann24/Fwaf-Machine-Learning-driven-Web-Application-Firewall/issues/8",
      "id": 328545995,
      "node_id": "MDU6SXNzdWUzMjg1NDU5OTU=",
      "number": 8,
      "title": "Script not working?",
      "user": {
        "login": "thlevy",
        "id": 34739079,
        "node_id": "MDQ6VXNlcjM0NzM5MDc5",
        "avatar_url": "https://avatars0.githubusercontent.com/u/34739079?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/thlevy",
        "html_url": "https://github.com/thlevy",
        "followers_url": "https://api.github.com/users/thlevy/followers",
        "following_url": "https://api.github.com/users/thlevy/following{/other_user}",
        "gists_url": "https://api.github.com/users/thlevy/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/thlevy/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/thlevy/subscriptions",
        "organizations_url": "https://api.github.com/users/thlevy/orgs",
        "repos_url": "https://api.github.com/users/thlevy/repos",
        "events_url": "https://api.github.com/users/thlevy/events{/privacy}",
        "received_events_url": "https://api.github.com/users/thlevy/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 2,
      "created_at": "2018-06-01T14:40:22Z",
      "updated_at": "2018-06-08T07:13:13Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "Hi,\r\nHas the ML script been run recently?\r\nI tried to run it with python 3 and 2 and got errors.\r\nFor inst with python 2 (might be 2.7.15), I got the following logs:\r\n_\"(py27) C:\\Users\\Thomas Lev\\Documents\\Fwaf\\Fwaf-Machine-Learning-driven-Web-Application-Firewall-master>python script.py\r\nC:\\Users\\Thomas Lev\\AppData\\Local\\conda\\conda\\envs\\py27\\lib\\site-packages\\sklearn\\cross_validation.py:41: DeprecationWarning: This module was deprecated in version 0.18 in favor of the model_selection module into which all the refactored classes and functions are moved. Also note that the interface of the new CV iterators are different from that of this module. This module will be removed in 0.20.\r\n  \"This module will be removed in 0.20.\", DeprecationWarning)\r\nTraceback (most recent call last):\r\n  File \"script.py\", line 43, in <module>\r\n    X = vectorizer.fit_transform(queries)\r\n  File \"C:\\Users\\Thomas Lev\\AppData\\Local\\conda\\conda\\envs\\py27\\lib\\site-packages\\sklearn\\feature_extraction\\text.py\", line 1381, in fit_transform\r\n    X = super(TfidfVectorizer, self).fit_transform(raw_documents)\r\n  File \"C:\\Users\\Thomas Lev\\AppData\\Local\\conda\\conda\\envs\\py27\\lib\\site-packages\\sklearn\\feature_extraction\\text.py\", line 869, in fit_transform\r\n    self.fixed_vocabulary_)\r\n  File \"C:\\Users\\Thomas Lev\\AppData\\Local\\conda\\conda\\envs\\py27\\lib\\site-packages\\sklearn\\feature_extraction\\text.py\", line 792, in _count_vocab\r\n    for feature in analyze(doc):\r\n  File \"C:\\Users\\Thomas Lev\\AppData\\Local\\conda\\conda\\envs\\py27\\lib\\site-packages\\sklearn\\feature_extraction\\text.py\", line 255, in <lambda>\r\n    return lambda doc: self._char_ngrams(preprocess(self.decode(doc)))\r\n  File \"C:\\Users\\Thomas Lev\\AppData\\Local\\conda\\conda\\envs\\py27\\lib\\site-packages\\sklearn\\feature_extraction\\text.py\", line 116, in decode\r\n    doc = doc.decode(self.encoding, self.decode_error)\r\n  File \"C:\\Users\\Thomas Lev\\AppData\\Local\\conda\\conda\\envs\\py27\\lib\\encodings\\utf_8.py\", line 16, in decode\r\n    return codecs.utf_8_decode(input, errors, True)\r\nUnicodeDecodeError: 'utf8' codec can't decode byte 0xc0 in position 1: invalid start byte\"_\r\n\r\nThanks\r\n\r\nThomas\r\n\r\n",
      "score": 68.223236
    },
    {
      "url": "https://api.github.com/repos/antoinecarme/pyaf/issues/92",
      "repository_url": "https://api.github.com/repos/antoinecarme/pyaf",
      "labels_url": "https://api.github.com/repos/antoinecarme/pyaf/issues/92/labels{/name}",
      "comments_url": "https://api.github.com/repos/antoinecarme/pyaf/issues/92/comments",
      "events_url": "https://api.github.com/repos/antoinecarme/pyaf/issues/92/events",
      "html_url": "https://github.com/antoinecarme/pyaf/pull/92",
      "id": 331681398,
      "node_id": "MDExOlB1bGxSZXF1ZXN0MTk0MzUyODcw",
      "number": 92,
      "title": "Make package directory and simplify package setup",
      "user": {
        "login": "jmabry",
        "id": 6393784,
        "node_id": "MDQ6VXNlcjYzOTM3ODQ=",
        "avatar_url": "https://avatars0.githubusercontent.com/u/6393784?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/jmabry",
        "html_url": "https://github.com/jmabry",
        "followers_url": "https://api.github.com/users/jmabry/followers",
        "following_url": "https://api.github.com/users/jmabry/following{/other_user}",
        "gists_url": "https://api.github.com/users/jmabry/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/jmabry/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/jmabry/subscriptions",
        "organizations_url": "https://api.github.com/users/jmabry/orgs",
        "repos_url": "https://api.github.com/users/jmabry/repos",
        "events_url": "https://api.github.com/users/jmabry/events{/privacy}",
        "received_events_url": "https://api.github.com/users/jmabry/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 1,
      "created_at": "2018-06-12T17:25:12Z",
      "updated_at": "2018-06-12T18:05:32Z",
      "closed_at": null,
      "author_association": "NONE",
      "pull_request": {
        "url": "https://api.github.com/repos/antoinecarme/pyaf/pulls/92",
        "html_url": "https://github.com/antoinecarme/pyaf/pull/92",
        "diff_url": "https://github.com/antoinecarme/pyaf/pull/92.diff",
        "patch_url": "https://github.com/antoinecarme/pyaf/pull/92.patch"
      },
      "body": "Closes #91 \r\n\r\nTested successfully with\r\n\r\n```\r\nconda create -n pyaf3\r\nconda activate pyaf3\r\nconda install pip\r\npip install scipy pandas sklearn matplotlib pydot dill pathos sqlalchemy\r\npip install -e .\r\npython tests/hierarchical/test_grouped.py\r\n```",
      "score": 66.97154
    },
    {
      "url": "https://api.github.com/repos/dmlc/xgboost/issues/3349",
      "repository_url": "https://api.github.com/repos/dmlc/xgboost",
      "labels_url": "https://api.github.com/repos/dmlc/xgboost/issues/3349/labels{/name}",
      "comments_url": "https://api.github.com/repos/dmlc/xgboost/issues/3349/comments",
      "events_url": "https://api.github.com/repos/dmlc/xgboost/issues/3349/events",
      "html_url": "https://github.com/dmlc/xgboost/issues/3349",
      "id": 327515013,
      "node_id": "MDU6SXNzdWUzMjc1MTUwMTM=",
      "number": 3349,
      "title": "base_score default for XGBModel",
      "user": {
        "login": "tluc4nala",
        "id": 39748395,
        "node_id": "MDQ6VXNlcjM5NzQ4Mzk1",
        "avatar_url": "https://avatars0.githubusercontent.com/u/39748395?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/tluc4nala",
        "html_url": "https://github.com/tluc4nala",
        "followers_url": "https://api.github.com/users/tluc4nala/followers",
        "following_url": "https://api.github.com/users/tluc4nala/following{/other_user}",
        "gists_url": "https://api.github.com/users/tluc4nala/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/tluc4nala/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/tluc4nala/subscriptions",
        "organizations_url": "https://api.github.com/users/tluc4nala/orgs",
        "repos_url": "https://api.github.com/users/tluc4nala/repos",
        "events_url": "https://api.github.com/users/tluc4nala/events{/privacy}",
        "received_events_url": "https://api.github.com/users/tluc4nala/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 1,
      "created_at": "2018-05-29T22:53:25Z",
      "updated_at": "2018-06-04T20:51:51Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "base_score should not default to 0.5 for regression problems.\r\nEither 0.0 or the mean of the response variable is more appropriate.\r\n\r\nhttps://github.com/dmlc/xgboost/blob/b7cbec4d4bf3e102164f083638d41de0aafa2999/python-package/xgboost/sklearn.py#L133",
      "score": 66.73894
    },
    {
      "url": "https://api.github.com/repos/Peter554/StainTools/issues/18",
      "repository_url": "https://api.github.com/repos/Peter554/StainTools",
      "labels_url": "https://api.github.com/repos/Peter554/StainTools/issues/18/labels{/name}",
      "comments_url": "https://api.github.com/repos/Peter554/StainTools/issues/18/comments",
      "events_url": "https://api.github.com/repos/Peter554/StainTools/issues/18/events",
      "html_url": "https://github.com/Peter554/StainTools/issues/18",
      "id": 328955377,
      "node_id": "MDU6SXNzdWUzMjg5NTUzNzc=",
      "number": 18,
      "title": "SPAMS dependency / is spams lasso the slowest part in the whole thing?",
      "user": {
        "login": "jgamper",
        "id": 13080433,
        "node_id": "MDQ6VXNlcjEzMDgwNDMz",
        "avatar_url": "https://avatars3.githubusercontent.com/u/13080433?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/jgamper",
        "html_url": "https://github.com/jgamper",
        "followers_url": "https://api.github.com/users/jgamper/followers",
        "following_url": "https://api.github.com/users/jgamper/following{/other_user}",
        "gists_url": "https://api.github.com/users/jgamper/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/jgamper/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/jgamper/subscriptions",
        "organizations_url": "https://api.github.com/users/jgamper/orgs",
        "repos_url": "https://api.github.com/users/jgamper/repos",
        "events_url": "https://api.github.com/users/jgamper/events{/privacy}",
        "received_events_url": "https://api.github.com/users/jgamper/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [
        {
          "id": 958603710,
          "node_id": "MDU6TGFiZWw5NTg2MDM3MTA=",
          "url": "https://api.github.com/repos/Peter554/StainTools/labels/discussion",
          "name": "discussion",
          "color": "ff00ff",
          "default": false
        }
      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 3,
      "created_at": "2018-06-04T08:26:32Z",
      "updated_at": "2018-06-08T09:25:22Z",
      "closed_at": null,
      "author_association": "COLLABORATOR",
      "body": "We could replace spams lasso with a small NN based mapping to the concentrations. That should speed up the augmentor. What do you think @Peter554 ?",
      "score": 66.39942
    },
    {
      "url": "https://api.github.com/repos/gtkfi/ArcSDM/issues/99",
      "repository_url": "https://api.github.com/repos/gtkfi/ArcSDM",
      "labels_url": "https://api.github.com/repos/gtkfi/ArcSDM/issues/99/labels{/name}",
      "comments_url": "https://api.github.com/repos/gtkfi/ArcSDM/issues/99/comments",
      "events_url": "https://api.github.com/repos/gtkfi/ArcSDM/issues/99/events",
      "html_url": "https://github.com/gtkfi/ArcSDM/issues/99",
      "id": 328098688,
      "node_id": "MDU6SXNzdWUzMjgwOTg2ODg=",
      "number": 99,
      "title": "Experimental toolbox doesn't load on 10.6",
      "user": {
        "login": "trotor",
        "id": 10844537,
        "node_id": "MDQ6VXNlcjEwODQ0NTM3",
        "avatar_url": "https://avatars0.githubusercontent.com/u/10844537?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/trotor",
        "html_url": "https://github.com/trotor",
        "followers_url": "https://api.github.com/users/trotor/followers",
        "following_url": "https://api.github.com/users/trotor/following{/other_user}",
        "gists_url": "https://api.github.com/users/trotor/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/trotor/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/trotor/subscriptions",
        "organizations_url": "https://api.github.com/users/trotor/orgs",
        "repos_url": "https://api.github.com/users/trotor/repos",
        "events_url": "https://api.github.com/users/trotor/events{/privacy}",
        "received_events_url": "https://api.github.com/users/trotor/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [
        {
          "id": 541757608,
          "node_id": "MDU6TGFiZWw1NDE3NTc2MDg=",
          "url": "https://api.github.com/repos/gtkfi/ArcSDM/labels/Waiting%20input",
          "name": "Waiting input",
          "color": "fbca04",
          "default": false
        },
        {
          "id": 351255517,
          "node_id": "MDU6TGFiZWwzNTEyNTU1MTc=",
          "url": "https://api.github.com/repos/gtkfi/ArcSDM/labels/help%20wanted",
          "name": "help wanted",
          "color": "128A0C",
          "default": true
        },
        {
          "id": 351255519,
          "node_id": "MDU6TGFiZWwzNTEyNTU1MTk=",
          "url": "https://api.github.com/repos/gtkfi/ArcSDM/labels/question",
          "name": "question",
          "color": "cc317c",
          "default": true
        }
      ],
      "state": "open",
      "locked": false,
      "assignee": {
        "login": "trotor",
        "id": 10844537,
        "node_id": "MDQ6VXNlcjEwODQ0NTM3",
        "avatar_url": "https://avatars0.githubusercontent.com/u/10844537?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/trotor",
        "html_url": "https://github.com/trotor",
        "followers_url": "https://api.github.com/users/trotor/followers",
        "following_url": "https://api.github.com/users/trotor/following{/other_user}",
        "gists_url": "https://api.github.com/users/trotor/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/trotor/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/trotor/subscriptions",
        "organizations_url": "https://api.github.com/users/trotor/orgs",
        "repos_url": "https://api.github.com/users/trotor/repos",
        "events_url": "https://api.github.com/users/trotor/events{/privacy}",
        "received_events_url": "https://api.github.com/users/trotor/received_events",
        "type": "User",
        "site_admin": false
      },
      "assignees": [
        {
          "login": "trotor",
          "id": 10844537,
          "node_id": "MDQ6VXNlcjEwODQ0NTM3",
          "avatar_url": "https://avatars0.githubusercontent.com/u/10844537?v=4",
          "gravatar_id": "",
          "url": "https://api.github.com/users/trotor",
          "html_url": "https://github.com/trotor",
          "followers_url": "https://api.github.com/users/trotor/followers",
          "following_url": "https://api.github.com/users/trotor/following{/other_user}",
          "gists_url": "https://api.github.com/users/trotor/gists{/gist_id}",
          "starred_url": "https://api.github.com/users/trotor/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/trotor/subscriptions",
          "organizations_url": "https://api.github.com/users/trotor/orgs",
          "repos_url": "https://api.github.com/users/trotor/repos",
          "events_url": "https://api.github.com/users/trotor/events{/privacy}",
          "received_events_url": "https://api.github.com/users/trotor/received_events",
          "type": "User",
          "site_admin": false
        }
      ],
      "milestone": null,
      "comments": 3,
      "created_at": "2018-05-31T11:20:24Z",
      "updated_at": "2018-06-11T14:01:00Z",
      "closed_at": null,
      "author_association": "MEMBER",
      "body": "There seems to be somekind of serious installation problem on 10.6 arcgis python vs. scikit-learn, sklearn and scipy\r\n\r\n![image](https://user-images.githubusercontent.com/10844537/40779449-c0048fde-64dd-11e8-8b82-0afc4d57a585.png)\r\n",
      "score": 66.19448
    },
    {
      "url": "https://api.github.com/repos/mvanga/pybootstrap/issues/2",
      "repository_url": "https://api.github.com/repos/mvanga/pybootstrap",
      "labels_url": "https://api.github.com/repos/mvanga/pybootstrap/issues/2/labels{/name}",
      "comments_url": "https://api.github.com/repos/mvanga/pybootstrap/issues/2/comments",
      "events_url": "https://api.github.com/repos/mvanga/pybootstrap/issues/2/events",
      "html_url": "https://github.com/mvanga/pybootstrap/issues/2",
      "id": 323249760,
      "node_id": "MDU6SXNzdWUzMjMyNDk3NjA=",
      "number": 2,
      "title": "Does not work in Python 2",
      "user": {
        "login": "fladd",
        "id": 2971539,
        "node_id": "MDQ6VXNlcjI5NzE1Mzk=",
        "avatar_url": "https://avatars3.githubusercontent.com/u/2971539?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/fladd",
        "html_url": "https://github.com/fladd",
        "followers_url": "https://api.github.com/users/fladd/followers",
        "following_url": "https://api.github.com/users/fladd/following{/other_user}",
        "gists_url": "https://api.github.com/users/fladd/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/fladd/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/fladd/subscriptions",
        "organizations_url": "https://api.github.com/users/fladd/orgs",
        "repos_url": "https://api.github.com/users/fladd/repos",
        "events_url": "https://api.github.com/users/fladd/events{/privacy}",
        "received_events_url": "https://api.github.com/users/fladd/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 0,
      "created_at": "2018-05-15T14:43:21Z",
      "updated_at": "2018-05-15T14:43:21Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "```\r\n  File \"/Library/Python/2.7/site-packages/pybootstrap/__init__.py\", line 54, in bootstrap\r\n    sample = sklearn.utils.resample(dataset, n_samples=n_size)\r\nAttributeError: 'module' object has no attribute 'utils'\r\n```",
      "score": 66.052925
    },
    {
      "url": "https://api.github.com/repos/Clinical-Genomics/development/issues/66",
      "repository_url": "https://api.github.com/repos/Clinical-Genomics/development",
      "labels_url": "https://api.github.com/repos/Clinical-Genomics/development/issues/66/labels{/name}",
      "comments_url": "https://api.github.com/repos/Clinical-Genomics/development/issues/66/comments",
      "events_url": "https://api.github.com/repos/Clinical-Genomics/development/issues/66/events",
      "html_url": "https://github.com/Clinical-Genomics/development/issues/66",
      "id": 332414967,
      "node_id": "MDU6SXNzdWUzMzI0MTQ5Njc=",
      "number": 66,
      "title": "Update svdb installation instructions",
      "user": {
        "login": "ingkebil",
        "id": 239608,
        "node_id": "MDQ6VXNlcjIzOTYwOA==",
        "avatar_url": "https://avatars3.githubusercontent.com/u/239608?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/ingkebil",
        "html_url": "https://github.com/ingkebil",
        "followers_url": "https://api.github.com/users/ingkebil/followers",
        "following_url": "https://api.github.com/users/ingkebil/following{/other_user}",
        "gists_url": "https://api.github.com/users/ingkebil/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/ingkebil/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/ingkebil/subscriptions",
        "organizations_url": "https://api.github.com/users/ingkebil/orgs",
        "repos_url": "https://api.github.com/users/ingkebil/repos",
        "events_url": "https://api.github.com/users/ingkebil/events{/privacy}",
        "received_events_url": "https://api.github.com/users/ingkebil/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": {
        "login": "henrikstranneheim",
        "id": 584738,
        "node_id": "MDQ6VXNlcjU4NDczOA==",
        "avatar_url": "https://avatars0.githubusercontent.com/u/584738?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/henrikstranneheim",
        "html_url": "https://github.com/henrikstranneheim",
        "followers_url": "https://api.github.com/users/henrikstranneheim/followers",
        "following_url": "https://api.github.com/users/henrikstranneheim/following{/other_user}",
        "gists_url": "https://api.github.com/users/henrikstranneheim/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/henrikstranneheim/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/henrikstranneheim/subscriptions",
        "organizations_url": "https://api.github.com/users/henrikstranneheim/orgs",
        "repos_url": "https://api.github.com/users/henrikstranneheim/repos",
        "events_url": "https://api.github.com/users/henrikstranneheim/events{/privacy}",
        "received_events_url": "https://api.github.com/users/henrikstranneheim/received_events",
        "type": "User",
        "site_admin": false
      },
      "assignees": [
        {
          "login": "henrikstranneheim",
          "id": 584738,
          "node_id": "MDQ6VXNlcjU4NDczOA==",
          "avatar_url": "https://avatars0.githubusercontent.com/u/584738?v=4",
          "gravatar_id": "",
          "url": "https://api.github.com/users/henrikstranneheim",
          "html_url": "https://github.com/henrikstranneheim",
          "followers_url": "https://api.github.com/users/henrikstranneheim/followers",
          "following_url": "https://api.github.com/users/henrikstranneheim/following{/other_user}",
          "gists_url": "https://api.github.com/users/henrikstranneheim/gists{/gist_id}",
          "starred_url": "https://api.github.com/users/henrikstranneheim/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/henrikstranneheim/subscriptions",
          "organizations_url": "https://api.github.com/users/henrikstranneheim/orgs",
          "repos_url": "https://api.github.com/users/henrikstranneheim/repos",
          "events_url": "https://api.github.com/users/henrikstranneheim/events{/privacy}",
          "received_events_url": "https://api.github.com/users/henrikstranneheim/received_events",
          "type": "User",
          "site_admin": false
        }
      ],
      "milestone": null,
      "comments": 0,
      "created_at": "2018-06-14T13:57:14Z",
      "updated_at": "2018-06-14T13:57:38Z",
      "closed_at": null,
      "author_association": "MEMBER",
      "body": "Update svdb installation instructions. See email conversation:\r\n\r\n@J35P312 says:\r\n```\r\n\r\nWoopsie daisy!\r\nThere seems to be some error with sklearn:\r\nImportError: cannot import name check_arrays\r\n\r\nEither you need to reinstall scikit learn, or you need to update to the latest svdb version (I dropped scikitlearn since it always end up giving these funky errors).\r\nI guess you got this error kinda recently? Have you changed you python environment in anyway? Maybe you can reset it?\r\n```\r\n\r\n@ingkebil says:\r\n```\r\nHi Jesper!\r\n\r\nthanks for the lightning response!\r\n\r\nI am trying to get MIP6 validated, so this particular conda env was not touched since february. We have already installed sklearn (which wasn't installed in the svdb env!)\r\n\r\nI'll install svdb from github then and try again. I'll keep you posted```\r\n",
      "score": 65.895584
    },
    {
      "url": "https://api.github.com/repos/openml/openml-python/issues/477",
      "repository_url": "https://api.github.com/repos/openml/openml-python",
      "labels_url": "https://api.github.com/repos/openml/openml-python/issues/477/labels{/name}",
      "comments_url": "https://api.github.com/repos/openml/openml-python/issues/477/comments",
      "events_url": "https://api.github.com/repos/openml/openml-python/issues/477/events",
      "html_url": "https://github.com/openml/openml-python/issues/477",
      "id": 329366398,
      "node_id": "MDU6SXNzdWUzMjkzNjYzOTg=",
      "number": 477,
      "title": "not JSON serializable error when running flow on a task",
      "user": {
        "login": "d-behi",
        "id": 2733801,
        "node_id": "MDQ6VXNlcjI3MzM4MDE=",
        "avatar_url": "https://avatars2.githubusercontent.com/u/2733801?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/d-behi",
        "html_url": "https://github.com/d-behi",
        "followers_url": "https://api.github.com/users/d-behi/followers",
        "following_url": "https://api.github.com/users/d-behi/following{/other_user}",
        "gists_url": "https://api.github.com/users/d-behi/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/d-behi/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/d-behi/subscriptions",
        "organizations_url": "https://api.github.com/users/d-behi/orgs",
        "repos_url": "https://api.github.com/users/d-behi/repos",
        "events_url": "https://api.github.com/users/d-behi/events{/privacy}",
        "received_events_url": "https://api.github.com/users/d-behi/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 7,
      "created_at": "2018-06-05T09:10:02Z",
      "updated_at": "2018-06-14T11:57:41Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "#### Description\r\nI encounter the following error when running a flow (for example 7103) on a task (for example 146607).\r\n`<openml.flows.flow.OpenMLFlow object at 0x11cc43110> is not JSON serializable`\r\n\r\n#### Steps/Code to Reproduce\r\n\r\nExample:\r\n```python\r\nimport openml\r\nflow = openml.flows.get_flow(7103)\r\ntask = openml.tasks.get_task(146607)\r\nrun = runs.run_flow_on_task(task, flow)\r\n```\r\n\r\n\r\n#### Expected Results\r\nOpenMLRun object\r\n\r\n#### Actual Results\r\n```\r\n---------------------------------------------------------------------------\r\nTypeError                                 Traceback (most recent call last)\r\n<ipython-input-102-f7fb112d9304> in <module>()\r\n----> 1 run = runs.run_flow_on_task(task, flow)\r\n\r\n/usr/local/lib/python2.7/site-packages/openml-0.7.0-py2.7.egg/openml/runs/functions.pyc in run_flow_on_task(task, flow, avoid_duplicate_runs, flow_tags, seed)\r\n     85     if avoid_duplicate_runs and flow_id:\r\n     86         flow_from_server = get_flow(flow_id)\r\n---> 87         setup_id = setup_exists(flow_from_server, flow.model)\r\n     88         ids = _run_exists(task.task_id, setup_id)\r\n     89         if ids:\r\n\r\n/usr/local/lib/python2.7/site-packages/openml-0.7.0-py2.7.egg/openml/setups/functions.pyc in setup_exists(flow, model)\r\n     44             raise ValueError('This should not happen!')\r\n     45 \r\n---> 46     openml_param_settings = openml.runs.OpenMLRun._parse_parameters(flow, model)\r\n     47     description = xmltodict.unparse(_to_dict(flow.flow_id,\r\n     48                                              openml_param_settings),\r\n\r\n/usr/local/lib/python2.7/site-packages/openml-0.7.0-py2.7.egg/openml/runs/run.pyc in _parse_parameters(flow, model)\r\n    360         flow_dict = get_flow_dict(flow)\r\n    361         parameters = extract_parameters(flow, flow_dict, model,\r\n--> 362                                         True, flow.flow_id)\r\n    363 \r\n    364         return parameters\r\n\r\n/usr/local/lib/python2.7/site-packages/openml-0.7.0-py2.7.egg/openml/runs/run.pyc in extract_parameters(_flow, _flow_dict, component_model, _main_call, main_id)\r\n    340                                 raise e\r\n    341                             elif not isinstance(step, openml.flows.OpenMLFlow):\r\n--> 342                                 raise e\r\n    343                         continue\r\n    344                     else:\r\n\r\nTypeError: <openml.flows.flow.OpenMLFlow object at 0x11cc43110> is not JSON serializable\r\n\r\n```\r\n\r\n#### Versions\r\nDarwin-17.5.0-x86_64-i386-64bit\r\n('Python', '2.7.15 (default, May 28 2018, 14:20:45) \\n[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.1)]')\r\n('NumPy', '1.13.3')\r\n('SciPy', '1.0.0')\r\n('Scikit-Learn', '0.18.1')\r\n('OpenML', '0.7.0')\r\n\r\n\r\n<!-- Thanks for contributing! -->",
      "score": 65.28625
    },
    {
      "url": "https://api.github.com/repos/RasaHQ/rasa_nlu/issues/1097",
      "repository_url": "https://api.github.com/repos/RasaHQ/rasa_nlu",
      "labels_url": "https://api.github.com/repos/RasaHQ/rasa_nlu/issues/1097/labels{/name}",
      "comments_url": "https://api.github.com/repos/RasaHQ/rasa_nlu/issues/1097/comments",
      "events_url": "https://api.github.com/repos/RasaHQ/rasa_nlu/issues/1097/events",
      "html_url": "https://github.com/RasaHQ/rasa_nlu/issues/1097",
      "id": 324867887,
      "node_id": "MDU6SXNzdWUzMjQ4Njc4ODc=",
      "number": 1097,
      "title": "Issue in migrating from dialogflow",
      "user": {
        "login": "Priprocks",
        "id": 24402389,
        "node_id": "MDQ6VXNlcjI0NDAyMzg5",
        "avatar_url": "https://avatars3.githubusercontent.com/u/24402389?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/Priprocks",
        "html_url": "https://github.com/Priprocks",
        "followers_url": "https://api.github.com/users/Priprocks/followers",
        "following_url": "https://api.github.com/users/Priprocks/following{/other_user}",
        "gists_url": "https://api.github.com/users/Priprocks/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/Priprocks/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/Priprocks/subscriptions",
        "organizations_url": "https://api.github.com/users/Priprocks/orgs",
        "repos_url": "https://api.github.com/users/Priprocks/repos",
        "events_url": "https://api.github.com/users/Priprocks/events{/privacy}",
        "received_events_url": "https://api.github.com/users/Priprocks/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [
        {
          "id": 461549567,
          "node_id": "MDU6TGFiZWw0NjE1NDk1Njc=",
          "url": "https://api.github.com/repos/RasaHQ/rasa_nlu/labels/question",
          "name": "question",
          "color": "cc317c",
          "default": true
        }
      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 2,
      "created_at": "2018-05-21T10:34:48Z",
      "updated_at": "2018-05-24T08:25:40Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "<!-- If you don't know your Rasa NLU version, here is some help: https://goo.gl/g9QQg2. If you are creating a feature request, feel free to remove all the system information stuff. --> \r\n\r\n**Rasa NLU version**:\r\n0.12.3\r\n**Operating system** (windows, osx, ...):\r\nWindows 10\r\n**Content of model configuration file**:\r\n```config_spacy.json\r\n{\r\n  \"pipeline\":\"spacy_sklearn\",\r\n  \"path\":\"./models/nlu\",\r\n  \"data\":\"./data/nlu\"\r\n}\r\n```\r\n\r\n**Issue**:\r\n\r\nI used the dialogflow folder given in the examples of the repo and named it nlu in ./data.\r\nFollowing the documentation of migrating dialogflow to rasa-core, I tried training the model by passing the 'nlu' directory as path.\r\nIt gave the following output:\r\n```\r\nC:\\My Projects\\Banking>python -m rasa_nlu.train -d ./data/nlu -c config_spacy.json --fixed_model_name bank_nlu\r\n2018-05-21 15:40:39 WARNING  rasa_nlu.training_data.formats.dialogflow - No training examples found for dialogflow file ./data/nlu\\intents\\Default Fallback Intent.json!\r\nFitting 2 folds for each of 6 candidates, totalling 12 fits\r\nC:\\Users\\pritamp533\\AppData\\Local\\Programs\\Python\\Python36\\lib\\site-packages\\sklearn\\metrics\\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.\r\n  'precision', 'predicted', average, warn_for)\r\nC:\\Users\\pritamp533\\AppData\\Local\\Programs\\Python\\Python36\\lib\\site-packages\\sklearn\\metrics\\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.\r\n  'precision', 'predicted', average, warn_for)\r\nC:\\Users\\pritamp533\\AppData\\Local\\Programs\\Python\\Python36\\lib\\site-packages\\sklearn\\metrics\\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.\r\n  'precision', 'predicted', average, warn_for)\r\nC:\\Users\\pritamp533\\AppData\\Local\\Programs\\Python\\Python36\\lib\\site-packages\\sklearn\\metrics\\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.\r\n  'precision', 'predicted', average, warn_for)\r\nC:\\Users\\pritamp533\\AppData\\Local\\Programs\\Python\\Python36\\lib\\site-packages\\sklearn\\metrics\\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.\r\n  'precision', 'predicted', average, warn_for)\r\nC:\\Users\\pritamp533\\AppData\\Local\\Programs\\Python\\Python36\\lib\\site-packages\\sklearn\\metrics\\classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.\r\n  'precision', 'predicted', average, warn_for)\r\n[Parallel(n_jobs=1)]: Done  12 out of  12 | elapsed:    0.0s finished\r\n```\r\n\r\n\r\nAnd after this, no model is generated. I am beginner and cannot understand the meaning of the above error. Please help!",
      "score": 65.040726
    },
    {
      "url": "https://api.github.com/repos/cmusatyalab/openface/issues/364",
      "repository_url": "https://api.github.com/repos/cmusatyalab/openface",
      "labels_url": "https://api.github.com/repos/cmusatyalab/openface/issues/364/labels{/name}",
      "comments_url": "https://api.github.com/repos/cmusatyalab/openface/issues/364/comments",
      "events_url": "https://api.github.com/repos/cmusatyalab/openface/issues/364/events",
      "html_url": "https://github.com/cmusatyalab/openface/issues/364",
      "id": 322682089,
      "node_id": "MDU6SXNzdWUzMjI2ODIwODk=",
      "number": 364,
      "title": "How do I fix this error?",
      "user": {
        "login": "emfls",
        "id": 23720257,
        "node_id": "MDQ6VXNlcjIzNzIwMjU3",
        "avatar_url": "https://avatars3.githubusercontent.com/u/23720257?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/emfls",
        "html_url": "https://github.com/emfls",
        "followers_url": "https://api.github.com/users/emfls/followers",
        "following_url": "https://api.github.com/users/emfls/following{/other_user}",
        "gists_url": "https://api.github.com/users/emfls/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/emfls/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/emfls/subscriptions",
        "organizations_url": "https://api.github.com/users/emfls/orgs",
        "repos_url": "https://api.github.com/users/emfls/repos",
        "events_url": "https://api.github.com/users/emfls/events{/privacy}",
        "received_events_url": "https://api.github.com/users/emfls/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 0,
      "created_at": "2018-05-14T06:00:10Z",
      "updated_at": "2018-05-14T06:00:10Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "openface baeg$ python3 ./demos/classifier.py train ./generated-embeddings/\r\n/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/sklearn/lda.py:6: DeprecationWarning: lda.LDA has been moved to discriminant_analysis.LinearDiscriminantAnalysis in 0.17 and will be removed in 0.19\r\n  \"in 0.17 and will be removed in 0.19\", DeprecationWarning)\r\n/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/sklearn/cross_validation.py:44: DeprecationWarning: This module was deprecated in version 0.18 in favor of the model_selection module into which all the refactored classes and functions are moved. Also note that the interface of the new CV iterators are different from that of this module. This module will be removed in 0.20.\r\n  \"This module will be removed in 0.20.\", DeprecationWarning)\r\n/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/sklearn/grid_search.py:43: DeprecationWarning: This module was deprecated in version 0.18 in favor of the model_selection module into which all the refactored classes and functions are moved. This module will be removed in 0.20.\r\n  DeprecationWarning)\r\nLoading embeddings.\r\nTraceback (most recent call last):\r\n  File \"./demos/classifier.py\", line 291, in <module>\r\n    train(args)\r\n  File \"./demos/classifier.py\", line 112, in train\r\n    le = LabelEncoder().fit(labels)\r\n  File \"/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/sklearn/preprocessing/label.py\", line 112, in fit\r\n    y = column_or_1d(y, warn=True)\r\n  File \"/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/sklearn/utils/validation.py\", line 562, in column_or_1d\r\n    raise ValueError(\"bad input shape {0}\".format(shape))\r\nValueError: bad input shape ()",
      "score": 64.29382
    },
    {
      "url": "https://api.github.com/repos/globality-corp/sklearn-hierarchical-classification/issues/2",
      "repository_url": "https://api.github.com/repos/globality-corp/sklearn-hierarchical-classification",
      "labels_url": "https://api.github.com/repos/globality-corp/sklearn-hierarchical-classification/issues/2/labels{/name}",
      "comments_url": "https://api.github.com/repos/globality-corp/sklearn-hierarchical-classification/issues/2/comments",
      "events_url": "https://api.github.com/repos/globality-corp/sklearn-hierarchical-classification/issues/2/events",
      "html_url": "https://github.com/globality-corp/sklearn-hierarchical-classification/issues/2",
      "id": 305434249,
      "node_id": "MDU6SXNzdWUzMDU0MzQyNDk=",
      "number": 2,
      "title": "Cannot install with pip",
      "user": {
        "login": "YannDubs",
        "id": 24327668,
        "node_id": "MDQ6VXNlcjI0MzI3NjY4",
        "avatar_url": "https://avatars3.githubusercontent.com/u/24327668?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/YannDubs",
        "html_url": "https://github.com/YannDubs",
        "followers_url": "https://api.github.com/users/YannDubs/followers",
        "following_url": "https://api.github.com/users/YannDubs/following{/other_user}",
        "gists_url": "https://api.github.com/users/YannDubs/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/YannDubs/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/YannDubs/subscriptions",
        "organizations_url": "https://api.github.com/users/YannDubs/orgs",
        "repos_url": "https://api.github.com/users/YannDubs/repos",
        "events_url": "https://api.github.com/users/YannDubs/events{/privacy}",
        "received_events_url": "https://api.github.com/users/YannDubs/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 6,
      "created_at": "2018-03-15T06:51:16Z",
      "updated_at": "2018-05-27T10:17:28Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "Hi,\r\n\r\nIt seems that `pip install sklearn-hierarchical-classification` doesn't work. tested on python 2 and 3, and I use a mac (although this shouldn't be relevant).\r\n\r\noutput :\r\n```\r\nCollecting sklearn-hierarchical-classification\r\n  Could not find a version that satisfies the requirement sklearn-hierarchical-classification (from versions: )\r\nNo matching distribution found for sklearn-hierarchical-classification\r\n```\r\n\r\nCheers",
      "score": 63.994305
    },
    {
      "url": "https://api.github.com/repos/drscotthawley/panotti/issues/35",
      "repository_url": "https://api.github.com/repos/drscotthawley/panotti",
      "labels_url": "https://api.github.com/repos/drscotthawley/panotti/issues/35/labels{/name}",
      "comments_url": "https://api.github.com/repos/drscotthawley/panotti/issues/35/comments",
      "events_url": "https://api.github.com/repos/drscotthawley/panotti/issues/35/events",
      "html_url": "https://github.com/drscotthawley/panotti/issues/35",
      "id": 317586060,
      "node_id": "MDU6SXNzdWUzMTc1ODYwNjA=",
      "number": 35,
      "title": "Error: \"Only one class present in y_true. ROC AUC score is not defined in that case\"",
      "user": {
        "login": "gllort",
        "id": 4455286,
        "node_id": "MDQ6VXNlcjQ0NTUyODY=",
        "avatar_url": "https://avatars0.githubusercontent.com/u/4455286?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/gllort",
        "html_url": "https://github.com/gllort",
        "followers_url": "https://api.github.com/users/gllort/followers",
        "following_url": "https://api.github.com/users/gllort/following{/other_user}",
        "gists_url": "https://api.github.com/users/gllort/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/gllort/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/gllort/subscriptions",
        "organizations_url": "https://api.github.com/users/gllort/orgs",
        "repos_url": "https://api.github.com/users/gllort/repos",
        "events_url": "https://api.github.com/users/gllort/events{/privacy}",
        "received_events_url": "https://api.github.com/users/gllort/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 0,
      "created_at": "2018-04-25T11:24:50Z",
      "updated_at": "2018-04-25T11:24:50Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "Running the eval_network script I got the following error that arises after the predict phase. What could possibly cause this error? Thanks in advance!\r\n\r\nRunning predict...\r\nCounting mistakes\r\n    Found 1846 total mistakes out of 16960 attempts\r\n      Mistakes by class:\r\n         ...\r\n         /* details of the mistakes follow */\r\n         ...\r\nMeasuring ROC...\r\n\r\n/apps/K80/PYTHON/3.6.3_ML/lib/python3.6/site-packages/sklearn/metrics/ranking.py:571: UndefinedMetricWarning: No positive samples in y_true, true positive value should be meaningless\r\n  UndefinedMetricWarning)\r\nTraceback (most recent call last):\r\n  File \"./eval_network.py\", line 135, in <module>\r\n    eval_network(weights_file=args.weights, classpath=args.classpath, batch_size=args.batch_size)\r\n  File \"./eval_network.py\", line 88, in eval_network\r\n    auc_score = roc_auc_score(Y_test, y_scores)\r\n  File \"/apps/K80/PYTHON/3.6.3_ML/lib/python3.6/site-packages/sklearn/metrics/ranking.py\", line 277, in roc_auc_score\r\n    sample_weight=sample_weight)\r\n  File \"/apps/K80/PYTHON/3.6.3_ML/lib/python3.6/site-packages/sklearn/metrics/base.py\", line 118, in _average_binary_score\r\n    sample_weight=score_weight)\r\n  File \"/apps/K80/PYTHON/3.6.3_ML/lib/python3.6/site-packages/sklearn/metrics/ranking.py\", line 268, in _binary_roc_auc_score\r\n    raise ValueError(\"Only one class present in y_true. ROC AUC score \"\r\nValueError: Only one class present in y_true. ROC AUC score is not defined in that case.\r\n",
      "score": 63.96017
    },
    {
      "url": "https://api.github.com/repos/jayjoshi64/FIFA-AI/issues/6",
      "repository_url": "https://api.github.com/repos/jayjoshi64/FIFA-AI",
      "labels_url": "https://api.github.com/repos/jayjoshi64/FIFA-AI/issues/6/labels{/name}",
      "comments_url": "https://api.github.com/repos/jayjoshi64/FIFA-AI/issues/6/comments",
      "events_url": "https://api.github.com/repos/jayjoshi64/FIFA-AI/issues/6/events",
      "html_url": "https://github.com/jayjoshi64/FIFA-AI/issues/6",
      "id": 308349010,
      "node_id": "MDU6SXNzdWUzMDgzNDkwMTA=",
      "number": 6,
      "title": "[Scikit-learn] : Description of Classes with it's useful methods which can be useful in implementation of Module Scikit-learn",
      "user": {
        "login": "jayjoshi64",
        "id": 20963493,
        "node_id": "MDQ6VXNlcjIwOTYzNDkz",
        "avatar_url": "https://avatars3.githubusercontent.com/u/20963493?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/jayjoshi64",
        "html_url": "https://github.com/jayjoshi64",
        "followers_url": "https://api.github.com/users/jayjoshi64/followers",
        "following_url": "https://api.github.com/users/jayjoshi64/following{/other_user}",
        "gists_url": "https://api.github.com/users/jayjoshi64/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/jayjoshi64/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/jayjoshi64/subscriptions",
        "organizations_url": "https://api.github.com/users/jayjoshi64/orgs",
        "repos_url": "https://api.github.com/users/jayjoshi64/repos",
        "events_url": "https://api.github.com/users/jayjoshi64/events{/privacy}",
        "received_events_url": "https://api.github.com/users/jayjoshi64/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [
        {
          "id": 878187723,
          "node_id": "MDU6TGFiZWw4NzgxODc3MjM=",
          "url": "https://api.github.com/repos/jayjoshi64/FIFA-AI/labels/Knowledge",
          "name": "Knowledge",
          "color": "1d76db",
          "default": false
        }
      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 0,
      "created_at": "2018-03-25T13:17:52Z",
      "updated_at": "2018-03-25T13:21:11Z",
      "closed_at": null,
      "author_association": "OWNER",
      "body": "Classes that will be covered in this issue:\r\n- [ ] sklearn.model_selection.train_test_split\r\n- [ ] sklearn.preprocessing.StandardScaler\r\n- [ ] sklearn.neural_network.* (All classes that can be useful)\r\n- [ ] sklearn.metrics.classification_report\r\n- [ ] sklearn.metrics.confusion_matrix\r\n\r\nAcceptance Criteria :\r\n\r\n- write description of class and purpose of it\r\n- Make a table in markdown language, that will cover the list of methods with it's description, purpose and parameters. And post it here and in a markdown file that will be stored in Learn/Python Modules\r\n",
      "score": 63.89732
    },
    {
      "url": "https://api.github.com/repos/mgmacias95/Flower-Recognition/issues/2",
      "repository_url": "https://api.github.com/repos/mgmacias95/Flower-Recognition",
      "labels_url": "https://api.github.com/repos/mgmacias95/Flower-Recognition/issues/2/labels{/name}",
      "comments_url": "https://api.github.com/repos/mgmacias95/Flower-Recognition/issues/2/comments",
      "events_url": "https://api.github.com/repos/mgmacias95/Flower-Recognition/issues/2/events",
      "html_url": "https://github.com/mgmacias95/Flower-Recognition/issues/2",
      "id": 318026062,
      "node_id": "MDU6SXNzdWUzMTgwMjYwNjI=",
      "number": 2,
      "title": "svm issue",
      "user": {
        "login": "Data-Scientist-Singh",
        "id": 37980233,
        "node_id": "MDQ6VXNlcjM3OTgwMjMz",
        "avatar_url": "https://avatars2.githubusercontent.com/u/37980233?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/Data-Scientist-Singh",
        "html_url": "https://github.com/Data-Scientist-Singh",
        "followers_url": "https://api.github.com/users/Data-Scientist-Singh/followers",
        "following_url": "https://api.github.com/users/Data-Scientist-Singh/following{/other_user}",
        "gists_url": "https://api.github.com/users/Data-Scientist-Singh/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/Data-Scientist-Singh/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/Data-Scientist-Singh/subscriptions",
        "organizations_url": "https://api.github.com/users/Data-Scientist-Singh/orgs",
        "repos_url": "https://api.github.com/users/Data-Scientist-Singh/repos",
        "events_url": "https://api.github.com/users/Data-Scientist-Singh/events{/privacy}",
        "received_events_url": "https://api.github.com/users/Data-Scientist-Singh/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 0,
      "created_at": "2018-04-26T13:18:46Z",
      "updated_at": "2018-04-26T13:18:46Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "Traceback (most recent call last):\r\n  File \"C:\\Users\\tanve\\Desktop\\Flower-Recognition-master\\model.py\", line 2, in <module>\r\n    from sklearn.svm import SVC\r\n  File \"C:\\Users\\tanve\\AppData\\Local\\Programs\\Python\\Python36\\lib\\site-packages\\sklearn\\__init__.py\", line 133, in <module>\r\n    from . import __check_build\r\nImportError: cannot import name '__check_build'",
      "score": 62.870758
    },
    {
      "url": "https://api.github.com/repos/crownpku/Information-Extraction-Chinese/issues/56",
      "repository_url": "https://api.github.com/repos/crownpku/Information-Extraction-Chinese",
      "labels_url": "https://api.github.com/repos/crownpku/Information-Extraction-Chinese/issues/56/labels{/name}",
      "comments_url": "https://api.github.com/repos/crownpku/Information-Extraction-Chinese/issues/56/comments",
      "events_url": "https://api.github.com/repos/crownpku/Information-Extraction-Chinese/issues/56/events",
      "html_url": "https://github.com/crownpku/Information-Extraction-Chinese/issues/56",
      "id": 320202904,
      "node_id": "MDU6SXNzdWUzMjAyMDI5MDQ=",
      "number": 56,
      "title": "test_GRU中执行main_for_evalution问题",
      "user": {
        "login": "DENGCPU",
        "id": 38373535,
        "node_id": "MDQ6VXNlcjM4MzczNTM1",
        "avatar_url": "https://avatars1.githubusercontent.com/u/38373535?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/DENGCPU",
        "html_url": "https://github.com/DENGCPU",
        "followers_url": "https://api.github.com/users/DENGCPU/followers",
        "following_url": "https://api.github.com/users/DENGCPU/following{/other_user}",
        "gists_url": "https://api.github.com/users/DENGCPU/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/DENGCPU/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/DENGCPU/subscriptions",
        "organizations_url": "https://api.github.com/users/DENGCPU/orgs",
        "repos_url": "https://api.github.com/users/DENGCPU/repos",
        "events_url": "https://api.github.com/users/DENGCPU/events{/privacy}",
        "received_events_url": "https://api.github.com/users/DENGCPU/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 1,
      "created_at": "2018-05-04T09:11:38Z",
      "updated_at": "2018-06-15T10:08:31Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "您好，我在运行test_GRU中的main_for_evalution 函数时，出现以下问题：\r\nFile \"/usr/local/lib/python3.5/dist-packages/tensorflow/python/platform/app.py\", line 124, in run\r\n    _sys.exit(main(argv))\r\n  File \"test_evaluation.py\", line 122, in main\r\n    average_precision = average_precision_score(allans, allprob)\r\n  File \"/usr/local/lib/python3.5/dist-packages/sklearn/metrics/ranking.py\", line 199, in average_precision_score\r\n    sample_weight=sample_weight)\r\n  File \"/usr/local/lib/python3.5/dist-packages/sklearn/metrics/base.py\", line 75, in _average_binary_score\r\n    return binary_metric(y_true, y_score, sample_weight=sample_weight)\r\n  File \"/usr/local/lib/python3.5/dist-packages/sklearn/metrics/ranking.py\", line 191, in _binary_uninterpolated_average_precision\r\n    y_true, y_score, sample_weight=sample_weight)\r\n  File \"/usr/local/lib/python3.5/dist-packages/sklearn/metrics/ranking.py\", line 441, in precision_recall_curve\r\n    sample_weight=sample_weight)\r\n  File \"/usr/local/lib/python3.5/dist-packages/sklearn/metrics/ranking.py\", line 320, in _binary_clf_curve\r\n    check_consistent_length(y_true, y_score, sample_weight)\r\n  File \"/usr/local/lib/python3.5/dist-packages/sklearn/utils/validation.py\", line 204, in check_consistent_length\r\n    \" samples: %r\" % [int(l) for l in lengths])\r\nValueError: Found input variables with inconsistent numbers of samples: [12008, 12000]\r\n不知道怎么解决，想咨询下您\r\n",
      "score": 62.46347
    },
    {
      "url": "https://api.github.com/repos/rsteca/sklearn-deap/issues/44",
      "repository_url": "https://api.github.com/repos/rsteca/sklearn-deap",
      "labels_url": "https://api.github.com/repos/rsteca/sklearn-deap/issues/44/labels{/name}",
      "comments_url": "https://api.github.com/repos/rsteca/sklearn-deap/issues/44/comments",
      "events_url": "https://api.github.com/repos/rsteca/sklearn-deap/issues/44/events",
      "html_url": "https://github.com/rsteca/sklearn-deap/issues/44",
      "id": 271011142,
      "node_id": "MDU6SXNzdWUyNzEwMTExNDI=",
      "number": 44,
      "title": "AttributeError: can't set attribute",
      "user": {
        "login": "vlatorre847",
        "id": 28892547,
        "node_id": "MDQ6VXNlcjI4ODkyNTQ3",
        "avatar_url": "https://avatars1.githubusercontent.com/u/28892547?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/vlatorre847",
        "html_url": "https://github.com/vlatorre847",
        "followers_url": "https://api.github.com/users/vlatorre847/followers",
        "following_url": "https://api.github.com/users/vlatorre847/following{/other_user}",
        "gists_url": "https://api.github.com/users/vlatorre847/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/vlatorre847/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/vlatorre847/subscriptions",
        "organizations_url": "https://api.github.com/users/vlatorre847/orgs",
        "repos_url": "https://api.github.com/users/vlatorre847/repos",
        "events_url": "https://api.github.com/users/vlatorre847/events{/privacy}",
        "received_events_url": "https://api.github.com/users/vlatorre847/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 8,
      "created_at": "2017-11-03T15:08:51Z",
      "updated_at": "2018-06-08T08:58:50Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "This is quite puzzling. \r\nmacos 10.9.5, \r\npython 3.5.2\r\nwith anaconda and spyder \r\n\r\nMy code:\r\n\r\n```\r\nimport sklearn\r\nimport numpy\r\nfrom sklearn.model_selection import KFold\r\nimport sklearn.neural_network\r\nimport sklearn.svm\r\nimport sklearn.ensemble\r\nimport sklearn.datasets\r\nimport time\r\nfrom evolutionary_search import EvolutionaryAlgorithmSearchCV\r\n\r\ndataset=sklearn.datasets.load_iris()\r\nX=dataset.data\r\nY=dataset.target\r\nseed=1\r\ntest_size = int( 0.2 * len( Y ) )\r\nnumpy.random.seed( seed )\r\nindices = numpy.random.permutation(len(X))\r\nX_train = X[ indices[:-test_size]]\r\nY_train = Y[ indices[:-test_size]]\r\nX_test = X[ indices[-test_size:]]\r\nY_test = Y[ indices[-test_size:]]\r\n\r\nnetwork=sklearn.svm.SVC()\r\nlb=[-15,-5]\r\nub=[3,15]\r\nspace_num=[]\r\nspace_points=[0]*len(lb)\r\n\r\nfor i in range(len(lb)):\r\n    space_num.append(int(ub[i]-lb[i]+1))    \r\n    space_points[i]=numpy.logspace(lb[i],ub[i],space_num[i],base=2)\r\nparam_grid={'gamma':space_points[0],'C':space_points[1]}\r\n\r\nscore=sklearn.metrics.make_scorer(sklearn.metrics.accuracy_score)\r\nclf=EvolutionaryAlgorithmSearchCV(network,params=param_grid,cv=3,scoring=score,verbose=2,n_jobs=1)\r\ntime_begin=time.time()\r\nclf.fit(X_train,Y_train)\r\ntime_end=time.time()-time_begin\r\n\r\nprint(time_end,sklearn.metrics.accuracy_score(clf.predict(X_test),Y_test))\r\n\r\n```\r\n\r\nI get the following error:\r\nFile \"~/anaconda/lib/python3.5/site-packages/evolutionary_search/cv.py\", line 301, in __init__\r\n    self.best_score_ = None\r\n\r\nAttributeError: can't set attribute\r\n\r\nAny idea?",
      "score": 62.0098
    },
    {
      "url": "https://api.github.com/repos/dmlc/xgboost/issues/2937",
      "repository_url": "https://api.github.com/repos/dmlc/xgboost",
      "labels_url": "https://api.github.com/repos/dmlc/xgboost/issues/2937/labels{/name}",
      "comments_url": "https://api.github.com/repos/dmlc/xgboost/issues/2937/comments",
      "events_url": "https://api.github.com/repos/dmlc/xgboost/issues/2937/events",
      "html_url": "https://github.com/dmlc/xgboost/pull/2937",
      "id": 280649274,
      "node_id": "MDExOlB1bGxSZXF1ZXN0MTU3MzU2ODg2",
      "number": 2937,
      "title": "Add grouping to Python sklearn interface",
      "user": {
        "login": "munro",
        "id": 500774,
        "node_id": "MDQ6VXNlcjUwMDc3NA==",
        "avatar_url": "https://avatars1.githubusercontent.com/u/500774?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/munro",
        "html_url": "https://github.com/munro",
        "followers_url": "https://api.github.com/users/munro/followers",
        "following_url": "https://api.github.com/users/munro/following{/other_user}",
        "gists_url": "https://api.github.com/users/munro/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/munro/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/munro/subscriptions",
        "organizations_url": "https://api.github.com/users/munro/orgs",
        "repos_url": "https://api.github.com/users/munro/repos",
        "events_url": "https://api.github.com/users/munro/events{/privacy}",
        "received_events_url": "https://api.github.com/users/munro/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 0,
      "created_at": "2017-12-08T23:41:05Z",
      "updated_at": "2018-01-25T19:16:21Z",
      "closed_at": null,
      "author_association": "NONE",
      "pull_request": {
        "url": "https://api.github.com/repos/dmlc/xgboost/pulls/2937",
        "html_url": "https://github.com/dmlc/xgboost/pull/2937",
        "diff_url": "https://github.com/dmlc/xgboost/pull/2937.diff",
        "patch_url": "https://github.com/dmlc/xgboost/pull/2937.patch"
      },
      "body": "Adds support to pass through `groups` & `group_sizes` to XGBoost's `sklearn` interface.\r\n\r\n~~This is more of a proof of concept, I want to rework the interface so that `groups` is used the way the `sklearn` does things.  With `sklearn` groups should be a group identifier, while for `xgboost` it expects the data to be sorted by group, and then given group sizes.  I was thinking about making a `group_sizes` parameter, and then having a `groups` parameter that works the `sklearn` way.  Let me know if that sounds good!~~\r\n\r\nThis adds ability to build ranking models through the `sklearn` API, by setting `.fit(X, y, groups=groups)`, which will now:\r\n1) resort the data so groups are next to each other (based on `groups`)\r\n2) on the created dmatrix, it will call `DMatrix.set_group(...)` with the calculated group sizes\r\n3) goes through existing `Booster.predict(...)` code path\r\n4) resorts the results back to what they original were\r\n5) returns back to what called the `sklearn` API\r\n\r\nI also passed through `group_sizes` if the programmer would rather use the XGBoost style of handing grouping, instead of how sklearn handles it (which is to assign a group number for every row).\r\n\r\n#815 seems related, but it looks like they just jumped down to using `DMatrix` & `train` directly... but I have a bunch of parameter tuning setup so I don't want to rework that code. 😄 ",
      "score": 61.678967
    },
    {
      "url": "https://api.github.com/repos/keyunluo/python-ffm/issues/4",
      "repository_url": "https://api.github.com/repos/keyunluo/python-ffm",
      "labels_url": "https://api.github.com/repos/keyunluo/python-ffm/issues/4/labels{/name}",
      "comments_url": "https://api.github.com/repos/keyunluo/python-ffm/issues/4/comments",
      "events_url": "https://api.github.com/repos/keyunluo/python-ffm/issues/4/events",
      "html_url": "https://github.com/keyunluo/python-ffm/issues/4",
      "id": 324081649,
      "node_id": "MDU6SXNzdWUzMjQwODE2NDk=",
      "number": 4,
      "title": "import error",
      "user": {
        "login": "lxin81",
        "id": 10075556,
        "node_id": "MDQ6VXNlcjEwMDc1NTU2",
        "avatar_url": "https://avatars2.githubusercontent.com/u/10075556?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/lxin81",
        "html_url": "https://github.com/lxin81",
        "followers_url": "https://api.github.com/users/lxin81/followers",
        "following_url": "https://api.github.com/users/lxin81/following{/other_user}",
        "gists_url": "https://api.github.com/users/lxin81/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/lxin81/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/lxin81/subscriptions",
        "organizations_url": "https://api.github.com/users/lxin81/orgs",
        "repos_url": "https://api.github.com/users/lxin81/repos",
        "events_url": "https://api.github.com/users/lxin81/events{/privacy}",
        "received_events_url": "https://api.github.com/users/lxin81/received_events",
        "type": "User",
        "site_admin": false
      },
      "labels": [

      ],
      "state": "open",
      "locked": false,
      "assignee": null,
      "assignees": [

      ],
      "milestone": null,
      "comments": 0,
      "created_at": "2018-05-17T15:46:32Z",
      "updated_at": "2018-05-17T15:46:32Z",
      "closed_at": null,
      "author_association": "NONE",
      "body": "Traceback (most recent call last):\r\n  File \"sklearn-example.py\", line 7, in <module>\r\n    from ffm import FFM, FFMFormatPandas, FFMData, save_data\r\n  File \"/media/lx/项目/tx/python-ffm/ffm/__init__.py\", line 1, in <module>\r\n    from .ffm import *\r\n  File \"/media/lx/项目/tx/python-ffm/ffm/ffm.py\", line 5, in <module>\r\n    libffm_file = [i for i in os.listdir(path) if i.startswith('libffm.')][0]\r\nIndexError: list index out of range\r\n",
      "score": 61.204494
    }
  ]
}
